<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUchicago" />

<meta name="date" content="2021-06-16" />

<title>Phylogenetic_strain_selection</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Kp_suppression</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/Kp_suppression">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Phylogenetic_strain_selection</h1>
<h4 class="author">KiseokUchicago</h4>
<h4 class="date">2021-06-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Kp_suppression/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210616code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210616)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210616code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210616)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoKpsuppressiontree0e83251924ad1c5101510d054bc57c9c53ea1b4etargetblank0e83251a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/Kp_suppression/tree/0e83251924ad1c5101510d054bc57c9c53ea1b4e" target="_blank">0e83251</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoKpsuppressiontree0e83251924ad1c5101510d054bc57c9c53ea1b4etargetblank0e83251a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/Kp_suppression/tree/0e83251924ad1c5101510d054bc57c9c53ea1b4e" target="_blank">0e83251</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  100per_umap.pdf
    Untracked:  1_Combinations_UMAP_Mix01-20.pdf
    Untracked:  20_strain_NRI_table.xlsx
    Untracked:  20_strain_NTI_table.xlsx
    Untracked:  220326_20_Community_members_UMAP_dispersal_maximization_KS.xlsx
    Untracked:  220326_20_Community_members_UMAP_dispersal_maximization_KS_formatted.xlsx
    Untracked:  220326_20_Community_members_UMAP_dispersal_maximization_KS_overlap_check.xlsx
    Untracked:  220327_20_Community_members_UMAP_dispersal_maximization_KS_from_46_Strain.xlsx
    Untracked:  220327_20strain_design/
    Untracked:  220514_Community_design_UMAP_distance_maximization_KS_from_Karna_84_Strain.xlsx
    Untracked:  220602_Community_design_UMAP_distance_maximization_KS_from_Karna_36_Strain.xlsx
    Untracked:  220602_Community_design_UMAP_distance_maximization_KS_from_Karna_36_Strain_alphabetical.xlsx
    Untracked:  220602_Community_design_UMAP_distance_maximization_KS_from_Karna_36_Strain_use_this_for_pipeting.xlsx
    Untracked:  220602_Community_design_UMAP_distance_maximization_KS_from_Karna_36_Strain_use_this_for_pipeting_formatted.xlsx
    Untracked:  220606_Community_design_UMAP_distance_maximization_KS_from_Karna_31_Strain_alphabetical.xlsx
    Untracked:  220606_Community_design_UMAP_distance_maximization_KS_from_Karna_31_Strain_use_this_for_pipeting.xlsx
    Untracked:  2_Combinations_UMAP_Mix20-41.pdf
    Untracked:  30per_umap.pdf
    Untracked:  31strains_design.pdf
    Untracked:  3_Combinations_UMAP_Mix41-60.pdf
    Untracked:  4_Combinations_UMAP_Mix61-80.pdf
    Untracked:  5_Combinations_UMAP_Mix81-96.pdf
    Untracked:  60per_umap.pdf
    Untracked:  96MemberCommunities-Rita.xlsx
    Untracked:  96_Community_maps_for_experiment_KS.xlsx
    Untracked:  96_Community_members_maximization_KS.xlsx
    Untracked:  96_Community_members_maximization_KS_saved.xlsx
    Untracked:  96_Community_members_maximization_KS_substituted.xlsx
    Untracked:  96_Community_members_maximization_KS_substituted_formatted.xlsx
    Untracked:  96_Community_members_maximization_KS_substituted_formatted_reordered_for_experiment.xlsx
    Untracked:  Community_design_UMAP.pdf
    Untracked:  Heatmap_overlap_ratio.pdf
    Untracked:  NRI_table.xlsx
    Untracked:  NTI_table.xlsx
    Untracked:  Rplot.pdf
    Untracked:  Rplot01.pdf
    Untracked:  Strain_selected_20_from_66_ver1.xlsx
    Untracked:  Strain_selected_20_from_66_ver2.xlsx
    Untracked:  Strain_selected_20_from_66_ver3.xlsx
    Untracked:  Strain_selected_20_kmeans_max_likelihood_distance.xlsx
    Untracked:  Strain_selected_20_max_NTI.xlsx
    Untracked:  Strain_selected_20_min_NTI.xlsx
    Untracked:  Strain_selected_95_max_NTI.xlsx
    Untracked:  Tree_UMAP_20.pdf
    Untracked:  Tree_both.pdf
    Untracked:  Tree_phy.pdf
    Untracked:  Tree_phy_20.pdf
    Untracked:  Tree_random_20.pdf
    Untracked:  Tree_species_95_selected.pdf
    Untracked:  Tree_strain66.pdf
    Untracked:  Tree_umap.pdf
    Untracked:  UMAP_100centroid_kmeans.pdf
    Untracked:  UMAP_95_species_selected.pdf
    Untracked:  UMAP_all.pdf
    Untracked:  UMAP_clustering_kmeans.pdf
    Untracked:  UMAP_ml_clustering_20.pdf
    Untracked:  UMAP_random_20.pdf
    Untracked:  UMAP_strain20_30percentile.pdf
    Untracked:  UMAP_strain20_50percentile.pdf
    Untracked:  UMAP_strain20_max_nearest.pdf
    Untracked:  UMAP_strain20_meax_mean_nearest.pdf
    Untracked:  UMAP_strain66.pdf
    Untracked:  UMAP_umapclustering_100.pdf
    Untracked:  UMAP_umapclustering_20.pdf
    Untracked:  WellPlate_maps_for_66strains_KS.xlsx
    Untracked:  WellPlate_maps_for_66strains_KS_position_added.xlsx
    Untracked:  data/220514_Community_design_UMAP_distance_maximization_KS_from_Karna_84_Strain.xlsx
    Untracked:  data/220602_Community_design_UMAP_distance_maximization_KS_from_Karna_36_Strain_alphabetical.xlsx
    Untracked:  data/220606_Community_design_UMAP_distance_maximization_KS_from_Karna_31_Strain_alphabetical.xlsx
    Untracked:  data/220606_Community_design_UMAP_distance_maximization_KS_from_Karna_31_Strain_use_this_for_pipeting.xlsx
    Untracked:  data/96MemberCommunities_formatted.xlsx
    Untracked:  data/96_Community_members_maximization_KS_substituted_formatted.xlsx
    Untracked:  data/96_Community_members_maximization_KS_substituted_formatted_reordered_for_experiment.xlsx
    Untracked:  data/Biobank_align_mafft.fasta.mldist.tsv
    Untracked:  data/Biobank_align_mafft.fasta.treefile.nwk
    Untracked:  data/Biobank_align_mafft.fasta.treefile_rooted.nwk
    Untracked:  data/Phylo_Dist_Ahmed.csv
    Untracked:  data/Rita_Matrix_Kp_CFUs_Communities_composition_updatedMarch26.xlsx
    Untracked:  data/Strain_selected_95_max_NTI_CEKannotated_updated.xlsx
    Untracked:  data/Strains_KOs_Cleaned.csv
    Untracked:  data/Strains_KOs_Cleaned_01.csv
    Untracked:  data/biobank.umap.16sSeqs.csv
    Untracked:  input_for_fasta_conversion.txt
    Untracked:  negative_control_UMAP_dispersal_method.pdf
    Untracked:  new_out_of_36strains.pdf
    Untracked:  out_of_36.pdf
    Untracked:  overlap_ratio_matrix.xlsx
    Untracked:  past_files/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Phylogenetic_strain_selection.Rmd</code>) and HTML (<code>docs/Phylogenetic_strain_selection.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Kp_suppression/1bed4e14bf9d0e13bafdf87fb36dd4e51c4013f7/docs/Phylogenetic_strain_selection.html" target="_blank">1bed4e1</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Kp_suppression/blob/01d43776ac7c019af4e11d28949d8d910a8615c7/analysis/Phylogenetic_strain_selection.Rmd" target="_blank">01d4377</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-19
</td>
<td>
fin
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Kp_suppression/88f6fc518c60e481de747ac80f29c6ca733f1c4c/docs/Phylogenetic_strain_selection.html" target="_blank">88f6fc5</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Kp_suppression/blob/2e96eca3c4540502680907cb49d2d96a03d69728/analysis/Phylogenetic_strain_selection.Rmd" target="_blank">2e96eca</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-18
</td>
<td>
almost
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Kp_suppression/b74f2e41b55372e446ae0145511f39c17512de42/docs/Phylogenetic_strain_selection.html" target="_blank">b74f2e4</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Kp_suppression/blob/a223d639641b9478ba976eb97141c62a8881ff18/analysis/Phylogenetic_strain_selection.Rmd" target="_blank">a223d63</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-17
</td>
<td>
working
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/Kp_suppression/fe0d21bb3aeda56e1a8e19f6d85b04230e300fe2/docs/Phylogenetic_strain_selection.html" target="_blank">fe0d21b</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/Kp_suppression/blob/309d09bbc94eb89cbbc6ae6a7c5bdf06846c0393/analysis/Phylogenetic_strain_selection.Rmd" target="_blank">309d09b</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-06-17
</td>
<td>
initiate
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="phylogenetic-tree-construction" class="section level2">
<h2>Phylogenetic tree construction</h2>
<p><strong>Kiseok Lee</strong></p>
<pre class="r"><code># libraries
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(vegan)
library(tidyverse)
library(magrittr)
library(readxl)
library(reshape2)
library(gtools)
library(devtools)
library(openxlsx)
library(ape)

## theme for ggplot
mytheme &lt;- theme_bw() + 
  theme(plot.title = element_text(size = 19,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10, family=&quot;serif&quot;))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(),panel.background=element_blank(),panel.border=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))
  
mytheme_2d &lt;- theme_bw() + 
  theme(plot.title = element_text(size = 19,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 17,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=13, family=&quot;serif&quot;))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank(),panel.background=element_blank(),plot.background=element_blank()) +
  theme(axis.ticks = element_line(size = 1.1))


# color collection
my_color_collection &lt;- c(
  &quot;#CBD588&quot;, &quot;#5F7FC7&quot;, &quot;orange&quot;, &quot;#AD6F3B&quot;, &quot;#673770&quot;, 
  &quot;#D14285&quot;, &quot;#652926&quot;, &quot;#C84248&quot;, &quot;#8569D5&quot;, &quot;#5E738F&quot;,
  &quot;#D1A33D&quot;, &quot;#8A7C64&quot;, &quot;#599861&quot;,&quot;#616163&quot;, &quot;#FFCDB2&quot;,
  &quot;#6D9F71&quot;, &quot;#242F40&quot;,
   &quot;#CCA43B&quot;, &quot;#F92A82&quot;, &quot;#ED7B84&quot;, &quot;#7EB77F&quot;, 
  &quot;#DEC4A1&quot;, &quot;#E5D1D0&quot;, &#39;#0E8482&#39;, &#39;#C9DAEA&#39;, &#39;#337357&#39;, 
  &#39;#95C623&#39;, &#39;#E55812&#39;, &#39;#04471C&#39;, &#39;#F2D7EE&#39;, &#39;#D3BCC0&#39;, 
  &#39;#A5668B&#39;, &#39;#69306D&#39;, &#39;#0E103D&#39;, &#39;#1A535C&#39;, &#39;#4ECDC4&#39;, 
  &#39;#F7FFF7&#39;, &#39;#FF6B6B&#39;, &#39;#FFE66D&#39;, &#39;#6699CC&#39;, &#39;#FFF275&#39;, 
  &#39;#FF8C42&#39;, &#39;#FF3C38&#39;, &#39;#A23E48&#39;, &#39;#000000&#39;, &#39;#CF5C36&#39;, 
  &#39;#EEE5E9&#39;, &#39;#7C7C7C&#39;, &#39;#EFC88B&#39;, &#39;#2E5266&#39;, &#39;#6E8898&#39;, 
  &#39;#9FB1BC&#39;, &#39;#D3D0CB&#39;, &#39;#E2C044&#39;, &#39;#5BC0EB&#39;, &#39;#FDE74C&#39;, 
  &#39;#9BC53D&#39;, &#39;#E55934&#39;, &#39;#FA7921&#39;, &quot;#CD9BCD&quot;, &quot;#508578&quot;, &quot;#DA5724&quot;)</code></pre>
</div>
<div id="data-input-and-pre-processing-for-phylogenetic-tree-construction" class="section level2">
<h2>1. Data input and pre-processing for phylogenetic tree construction</h2>
<pre class="r"><code>df_Biobank &lt;- read.csv(&#39;data/biobank.umap.16sSeqs.csv&#39;)

dim(df_Biobank)</code></pre>
<pre><code>[1] 850  13</code></pre>
<pre class="r"><code>colnames(df_Biobank)</code></pre>
<pre><code> [1] &quot;seq_id&quot;       &quot;umapA1&quot;       &quot;umapA2&quot;       &quot;tiplabel&quot;     &quot;msk_id&quot;      
 [6] &quot;length_bp&quot;    &quot;phylum&quot;       &quot;class&quot;        &quot;order&quot;        &quot;family&quot;      
[11] &quot;genus&quot;        &quot;species&quot;      &quot;nuc_sequence&quot;</code></pre>
<pre class="r"><code>head(df_Biobank)</code></pre>
<pre><code>             seq_id     umapA1     umapA2   tiplabel    msk_id length_bp
1   210119-DFI.3.32 -5.2351763  -4.934942 JGLPCKFE_0  DFI.3.32      1443
2   210119-DFI.3.61 -0.8318949 -11.305224 KFNOGCAE_0  DFI.3.61      1525
3 210119-DFI.3.86.2  0.2015703 -11.130230 MCNCDFEJ_0  DFI.3.86      1527
4  210119-DFI.4.132  1.4983046 -10.842872 EOGPJFFH_0 DFI.4.132      1528
5  210119-DFI.4.155 -1.4185973   8.737071 IKKMFIBH_0 DFI.4.155      1529
6   210119-DFI.4.81  1.4445846 -10.807266 GCOEKEKI_0  DFI.4.81      1124
          phylum               class            order             family
1 Proteobacteria Gammaproteobacteria Enterobacterales Enterobacteriaceae
2  Bacteroidetes         Bacteroidia    Bacteroidales     Bacteroidaceae
3  Bacteroidetes         Bacteroidia    Bacteroidales     Tannerellaceae
4  Bacteroidetes         Bacteroidia    Bacteroidales     Bacteroidaceae
5     Firmicutes          Clostridia    Clostridiales    Lachnospiraceae
6  Bacteroidetes         Bacteroidia    Bacteroidales     Bacteroidaceae
            genus                      species
1     Escherichia       Escherichia fergusonii
2     Bacteroides Bacteroides thetaiotaomicron
3 Parabacteroides    Parabacteroides johnsonii
4     Bacteroides            Bacteroides dorei
5    Anaerostipes          Anaerostipes hadrus
6     Bacteroides         Bacteroides vulgatus
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               nuc_sequence
1                                                                                       CGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT
2     ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATTTCAGTTTGCTTGCAAACTGGAGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGATAACTCGGGGATAGCCTTTCGAAAGAAAGATTAATACCCGATGGTATAATTAGACCGCATGGTCTTGTTATTAAAGAATTTCGGTTATCGATGGGGATGCGTTCCATTAGGCAGTTGGTGAGGTAACGGCTCACCAAACCTTCGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGCAGGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATATGGGAATAAAGTTTTCCACGTGTGGAATTTTGTATGTACCATATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGGTGGACAGTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGCTGTCTTGAGTACAGTAGAGGTGGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCTCACTGGACTGCAACTGACACTGATGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACAGTAAACGATGAATACTCGCTGTTTGCGATATACAGTAAGCGGCCAAGCGAAAGCATTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCATTTGAATATATTGGAAACAGTATAGTCGTAAGACAAATGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTATCTTTAGTTACTAACAGGTCATGCTGAGGACTCTAGAGAGACTGCCGTCGTAAGATGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACGTCCGGGGCTACACACGTGTTACAATGGGGGGTACAGAAGGCAGCTACCTGGTGACAGGATGCTAATCCCAAAAGCCTCTCTCAGTTCGGATCGAAGTCTGCAACCCGACTTCGTGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCATGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGAAAGCCGGGGGTACCTGAAGTACGTAACCGCAAGGAGCGTCCTAGGGTAAAACTGGTAATTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTT
3   CGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCGACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTAGGTAGCAATACTTATTGATGGCGACCGGCGCACGGGTGAGTAACGCGTATGCAACTTACCTATCAGAGGAGGATAGCCCGGCGAAAGTCGGATTAATACTCCATAAAACAGGGATCCCGCATGGGAATATTTGTTAAAGATTCATCGCTGATAGATAGGCATGCGTTCCATTAGGCAGTTGGCGGGGTAACGGCCCACCAAACCGACGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGTACTGAGACACGGACCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAAGGATGAAGGATCTATGGTTTGTAAACTTCTTTTATAGGGGAATAAAGTGTGGGACGTGTCCCATTTTGTATGTACCCTATGAATAAGCATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGTGGTAATTTAAGTCAGCGGTGAAAGTTTGTGGCTCAACCATAAAATTGCCGTTGAAACTGGGTTACTTGAGTGTGTTTGAGGTAGGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACTCCAATTGCGAAGGCAGCTTACTAAACCATAACTGACACTGAAGCACGAAAGCGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGATTACTAGGAGTTTGCGATAGACAGTAAGCTCTACAGCGAAAGCGTTAAGTAATCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGTTTGAACGTAGTCAGACCGACCTTGAAAGAGGTCTTCTAGCAATAGCTGATTACGAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTATCACTAGTTACTAACAGGTTAAGCTGAGGACTCTGGTGAGACTGCCAGCGTAAGCTGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACATCCGGGGCGACACACGTGTTACAATGGCATGGACAAAGGGCAGCTACCTGGCGACAGGATGCTAATCTCTAAACCATGTCTCAGTTCGGATCGGAGTCTGCAACTCGACTCCGTGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCATGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGGGAGCCGGGGGTACCTGAAGTCCGTAACCGCAAGGATCGGCCTAGGGTAAAACTGGTGACTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTTT
4  ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTCTTAGCTTGCTAAGGCTGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGTCTACTCTTGGCCAGCCTTCTGAAAGGAAGATTAATCCAGGATGGGATCATGAGTTCACATGTCCGCATGATTAAAGGTATTTTCCGGTAGACGATGGGGATGCGTTCCATTAGATAGTAGGCGGGGTAACGGCCCACCTAGTCAACGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGATGGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATAAAGGAATAAAGTCGGGTATGCATACCCGTTTGCATGTACTTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGATGGATGTTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGATGTCTTGAGTGCAGTTGAGGCAGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCCTGCTAAGCTGCAACTGACATTGAGGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACGGTAAACGATGAATACTCGCTGTTTGCGATATACGGCAAGCGGCCAAGCGAAAGCGTTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCACTCGAATGATCCGGAAACGGTTCAGCTAGCAATAGCGAGTGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTGTTGTCAGTTACTAACAGGTGATGCTGAGGACTCTGACAAGACTGCCATCGTAAGATGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACGTCCGGGGCTACACACGTGTTACAATGGGGGGTACAGAGGGCCGCTACCACGCGAGTGGATGCCAATCCCTAAAACCCCTCTCAGTTCGGACTGGAGTCTGCAACCCGACTCCACGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCACGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGGGAGCCGGGGGTACCTGAAGTGCGTAACCGCGAGGATCGCCCTAGGGTAAAACTGGTGACTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTT
5 ATGAGAGTTTGATCCTGGCTCAGGATGAACGCTGGCGGCGTGCTTAACACATGCAAGTCGAACGAAACACCTTATTTGATTTTCTTCGGAACTGAAGATTTGGTGATTGAGTGGCGGACGGGTGAGTAACGCGTGGGTAACCTGCCCTGTACAGGGGGATAACAGTCAGAAATGACTGCTAATACCGCATAAGACCACAGCACCGCATGGTGCAGGGGTAAAAACTCCGGTGGTACAGGATGGACCCGCGTCTGATTAGCTGGTTGGTGAGGTAACGGCTCACCAAGGCGACGATCAGTAGCCGGCTTGAGAGAGTGAACGGCCACATTGGGACTGAGACACGGCCCAAACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAGTGAAGAAGTATCTCGGTATGTAAAGCTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGTGCGTAGGTGGTATGGCAAGTCAGAAGTGAAAACCCAGGGCTTAACTCTGGGACTGCTTTTGAAACTGTCAGACTAGAGTGCAGGAGAGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACATCAGTGGCGAAGGCGGCTTACTGGACTGAAACTGACACTGAGGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGAATACTAGGTGTCGGGGCCGTAGAGGCTTCGGTGCCGCAGCCAACGCAGTAAGTATTCCACCTGGGGAGTACGTTCGCAAGAATGAAACTCAAAGGAATTGACGGGGACCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTTACCTGGTCTTGACATCCTTCTGACCGGTCCTTAACCGGACCTTTCCTTCGGGACAGGAGTGACAGGTGGTGCATGGTTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCCTATCTTTAGTAGCCAGCATATAAGGTGGGCACTCTAGAGAGACTGCCAGGGATAACCTGGAGGAAGGTGGGGACGACGTCAAATCATCATGCCCCTTATGACCAGGGCTACACACGTGCTACAATGGCGTAAACAGAGGGAAGCAGCCTCGTGAGAGTGAGCAAATCCCAAAAATAACGTCTCAGTTCGGATTGTAGTCTGCAACTCGACTACATGAAGCTGGAATCGCTAGTAATCGCGAATCAGAATGTCGCGGTGAATACGTTCCCGGGTCTTGTACACACCGCCCGTCACACCATGGGAGTCAGTAACGCCCGAAGTCAGTGACCCAACCGTAAGGAGGGAGCTGCCGAAGGCGGGACCGATAACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTT
6                                                                                                                                                                                                                                                                                                                                                                                                                      ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTCTTAGCTTGCTAAGGCCGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGTCTACTCTTGGACAGCCTTCTGAAAGGAAGATTAATACAAGATGGCATCATGAGTCCGCATGTTCACATGATTAAAGGTATTCCGGTAGACGATGGGGATGCGTTCCATTAGATAGTAGGCGGGGTAACGGCCCACCTAGTCTTCGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGAGAGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATAAAGGAATAAAGTCGGGTATGCATACCCGTTTGCATGTACTTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGATGGATGTTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGATATCTTGAGTGCAGTTGAGGCAGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCCTGCTAAGCTGCAACTGACATTGAGGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACGGTAAACGATGAATACTCGCTGTTTGCGATATACGGCAAGCGGCCAAGCGAAAGCGTTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCAGATGAATTACGGTGAAAGCCGTAAGCCGCAAGGCATCTGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTGTTGTCAGTTACTAACA</code></pre>
<pre class="r"><code># how many species
df_species &lt;- df_Biobank %&gt;% group_by(species) %&gt;% summarise(number=n_distinct(species))
dim(df_species)</code></pre>
<pre><code>[1] 117   2</code></pre>
<pre class="r"><code>head(df_species)</code></pre>
<pre><code># A tibble: 6 x 2
  species                       number
  &lt;chr&gt;                          &lt;int&gt;
1 [Clostridium] aldenense            1
2 [Clostridium] celerecrescens       1
3 [Clostridium] clostridioforme      1
4 [Clostridium] innocuum             1
5 [Clostridium] scindens             1
6 [Clostridium] symbiosum            1</code></pre>
<pre class="r"><code># get id and sequence
# id is seq_id + species
phylo_Biobank &lt;- df_Biobank %&gt;% unite(phy_id,c(&quot;seq_id&quot;,&#39;species&#39;),sep=&quot;_&quot;, remove=F)
phylo_Biobank$species &lt;- gsub(&quot;\\[&quot;,&quot;&quot;,phylo_Biobank$species)
phylo_Biobank$species &lt;- gsub(&quot;]&quot;,&quot;&quot;,phylo_Biobank$species)
colnames(phylo_Biobank)</code></pre>
<pre><code> [1] &quot;phy_id&quot;       &quot;seq_id&quot;       &quot;umapA1&quot;       &quot;umapA2&quot;       &quot;tiplabel&quot;    
 [6] &quot;msk_id&quot;       &quot;length_bp&quot;    &quot;phylum&quot;       &quot;class&quot;        &quot;order&quot;       
[11] &quot;family&quot;       &quot;genus&quot;        &quot;species&quot;      &quot;nuc_sequence&quot;</code></pre>
<pre class="r"><code># remove space in the id for downstream
phylo_Biobank$phy_id &lt;- gsub(&quot; &quot;,&quot;_&quot;,phylo_Biobank$phy_id) 
# remove &quot;[&quot; and &quot;]&quot;
phylo_Biobank$phy_id &lt;- gsub(&quot;\\[&quot;,&quot;&quot;,phylo_Biobank$phy_id)
phylo_Biobank$phy_id &lt;- gsub(&quot;]&quot;,&quot;&quot;,phylo_Biobank$phy_id)
col2_Biobank &lt;- phylo_Biobank %&gt;% dplyr::select(phy_id,nuc_sequence)

dim(col2_Biobank)</code></pre>
<pre><code>[1] 850   2</code></pre>
<pre class="r"><code>head(col2_Biobank)</code></pre>
<pre><code>                                        phy_id
1       210119-DFI.3.32_Escherichia_fergusonii
2 210119-DFI.3.61_Bacteroides_thetaiotaomicron
3  210119-DFI.3.86.2_Parabacteroides_johnsonii
4           210119-DFI.4.132_Bacteroides_dorei
5         210119-DFI.4.155_Anaerostipes_hadrus
6         210119-DFI.4.81_Bacteroides_vulgatus
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               nuc_sequence
1                                                                                       CGAGTGGCGGACGGGTGAGTAATGTCTGGGAAACTGCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATAACGTCGCAAGACCAAAGAGGGGGACCTTCGGGCCTCTTGCCATCGGATGTGCCCAGATGGGATTAGCTAGTAGGTGGGGTAACGGCTCACCTAGGCGACGATCCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGTCCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAAGCCTGATGCAGCCATGCCGCGTGTATGAAGAAGGCCTTCGGGTTGTAAAGTACTTTCAGCGGGGAGGAAGGGAGTAAAGTTAATACCTTTGCTCATTGACGTTACCCGCAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGTTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATCTGATACTGGCAAGCTTGAGTCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACGAAGACTGACGCTCAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGTCGACTTGGAGGTTGTGCCCTTGAGGCGTGGCTTCCGGAGCTAACGCGTTAAGTCGACCGCCTGGGGAGTACGGCCGCAAGGTTAAAACTCAAATGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGATGCAACGCGAAGAACCTTACCTGGTCTTGACATCCACGGAAGTTTTCAGAGATGAGAATGTGCCTTCGGGAACCGTGAGACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTTGTGAAATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTATCCTTTGTTGCCAGCGGTCCGGCCGGGAACTCAAAGGAGACTGCCAGTGATAAACTGGAGGAAGGTGGGGATGACGTCAAGTCATCATGGCCCTTACGACCAGGGCTACACACGTGCTACAATGGCGCATACAAAGAGAAGCGACCTCGCGAGAGCAAGCGGACCTCATAAAGTGCGTCGTAGTCCGGATTGGAGTCTGCAACTCGACTCCATGAAGTCGGAATCGCTAGTAATCGTGGATCAGAATGCCACGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCATGGGAGTGGGTTGCAAAAGAAGTAGGTAGCTTAACCTTCGGGAGGGCGCTTACCACTTTGTGATTCATGACTGGGGTGAAGTCGTAACAAGGTAACCGTAGGGGAACCTGCGGTTGGATCACCTCCTT
2     ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATTTCAGTTTGCTTGCAAACTGGAGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGATAACTCGGGGATAGCCTTTCGAAAGAAAGATTAATACCCGATGGTATAATTAGACCGCATGGTCTTGTTATTAAAGAATTTCGGTTATCGATGGGGATGCGTTCCATTAGGCAGTTGGTGAGGTAACGGCTCACCAAACCTTCGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGCAGGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATATGGGAATAAAGTTTTCCACGTGTGGAATTTTGTATGTACCATATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGGTGGACAGTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGCTGTCTTGAGTACAGTAGAGGTGGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCTCACTGGACTGCAACTGACACTGATGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACAGTAAACGATGAATACTCGCTGTTTGCGATATACAGTAAGCGGCCAAGCGAAAGCATTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCATTTGAATATATTGGAAACAGTATAGTCGTAAGACAAATGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTATCTTTAGTTACTAACAGGTCATGCTGAGGACTCTAGAGAGACTGCCGTCGTAAGATGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACGTCCGGGGCTACACACGTGTTACAATGGGGGGTACAGAAGGCAGCTACCTGGTGACAGGATGCTAATCCCAAAAGCCTCTCTCAGTTCGGATCGAAGTCTGCAACCCGACTTCGTGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCATGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGAAAGCCGGGGGTACCTGAAGTACGTAACCGCAAGGAGCGTCCTAGGGTAAAACTGGTAATTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTT
3   CGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCGACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTAGGTAGCAATACTTATTGATGGCGACCGGCGCACGGGTGAGTAACGCGTATGCAACTTACCTATCAGAGGAGGATAGCCCGGCGAAAGTCGGATTAATACTCCATAAAACAGGGATCCCGCATGGGAATATTTGTTAAAGATTCATCGCTGATAGATAGGCATGCGTTCCATTAGGCAGTTGGCGGGGTAACGGCCCACCAAACCGACGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGTACTGAGACACGGACCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGCCGAGAGGCTGAACCAGCCAAGTCGCGTGAAGGATGAAGGATCTATGGTTTGTAAACTTCTTTTATAGGGGAATAAAGTGTGGGACGTGTCCCATTTTGTATGTACCCTATGAATAAGCATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGTGGTAATTTAAGTCAGCGGTGAAAGTTTGTGGCTCAACCATAAAATTGCCGTTGAAACTGGGTTACTTGAGTGTGTTTGAGGTAGGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACTCCAATTGCGAAGGCAGCTTACTAAACCATAACTGACACTGAAGCACGAAAGCGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGATTACTAGGAGTTTGCGATAGACAGTAAGCTCTACAGCGAAAGCGTTAAGTAATCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGTTTGAACGTAGTCAGACCGACCTTGAAAGAGGTCTTCTAGCAATAGCTGATTACGAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTATCACTAGTTACTAACAGGTTAAGCTGAGGACTCTGGTGAGACTGCCAGCGTAAGCTGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACATCCGGGGCGACACACGTGTTACAATGGCATGGACAAAGGGCAGCTACCTGGCGACAGGATGCTAATCTCTAAACCATGTCTCAGTTCGGATCGGAGTCTGCAACTCGACTCCGTGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCATGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGGGAGCCGGGGGTACCTGAAGTCCGTAACCGCAAGGATCGGCCTAGGGTAAAACTGGTGACTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTTT
4  ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTCTTAGCTTGCTAAGGCTGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGTCTACTCTTGGCCAGCCTTCTGAAAGGAAGATTAATCCAGGATGGGATCATGAGTTCACATGTCCGCATGATTAAAGGTATTTTCCGGTAGACGATGGGGATGCGTTCCATTAGATAGTAGGCGGGGTAACGGCCCACCTAGTCAACGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGATGGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATAAAGGAATAAAGTCGGGTATGCATACCCGTTTGCATGTACTTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGATGGATGTTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGATGTCTTGAGTGCAGTTGAGGCAGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCCTGCTAAGCTGCAACTGACATTGAGGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACGGTAAACGATGAATACTCGCTGTTTGCGATATACGGCAAGCGGCCAAGCGAAAGCGTTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCACTCGAATGATCCGGAAACGGTTCAGCTAGCAATAGCGAGTGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTGTTGTCAGTTACTAACAGGTGATGCTGAGGACTCTGACAAGACTGCCATCGTAAGATGTGAGGAAGGTGGGGATGACGTCAAATCAGCACGGCCCTTACGTCCGGGGCTACACACGTGTTACAATGGGGGGTACAGAGGGCCGCTACCACGCGAGTGGATGCCAATCCCTAAAACCCCTCTCAGTTCGGACTGGAGTCTGCAACCCGACTCCACGAAGCTGGATTCGCTAGTAATCGCGCATCAGCCACGGCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCAAGCCATGGGAGCCGGGGGTACCTGAAGTGCGTAACCGCGAGGATCGCCCTAGGGTAAAACTGGTGACTGGGGCTAAGTCGTAACAAGGTAGCCGTACCGGAAGGTGCGGCTGGAACACCTCCTT
5 ATGAGAGTTTGATCCTGGCTCAGGATGAACGCTGGCGGCGTGCTTAACACATGCAAGTCGAACGAAACACCTTATTTGATTTTCTTCGGAACTGAAGATTTGGTGATTGAGTGGCGGACGGGTGAGTAACGCGTGGGTAACCTGCCCTGTACAGGGGGATAACAGTCAGAAATGACTGCTAATACCGCATAAGACCACAGCACCGCATGGTGCAGGGGTAAAAACTCCGGTGGTACAGGATGGACCCGCGTCTGATTAGCTGGTTGGTGAGGTAACGGCTCACCAAGGCGACGATCAGTAGCCGGCTTGAGAGAGTGAACGGCCACATTGGGACTGAGACACGGCCCAAACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGGGAAACCCTGATGCAGCGACGCCGCGTGAGTGAAGAAGTATCTCGGTATGTAAAGCTCTATCAGCAGGGAAGAAAATGACGGTACCTGACTAAGAAGCCCCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGTGCGTAGGTGGTATGGCAAGTCAGAAGTGAAAACCCAGGGCTTAACTCTGGGACTGCTTTTGAAACTGTCAGACTAGAGTGCAGGAGAGGTAAGCGGAATTCCTAGTGTAGCGGTGAAATGCGTAGATATTAGGAGGAACATCAGTGGCGAAGGCGGCTTACTGGACTGAAACTGACACTGAGGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGAATACTAGGTGTCGGGGCCGTAGAGGCTTCGGTGCCGCAGCCAACGCAGTAAGTATTCCACCTGGGGAGTACGTTCGCAAGAATGAAACTCAAAGGAATTGACGGGGACCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTTACCTGGTCTTGACATCCTTCTGACCGGTCCTTAACCGGACCTTTCCTTCGGGACAGGAGTGACAGGTGGTGCATGGTTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCCTATCTTTAGTAGCCAGCATATAAGGTGGGCACTCTAGAGAGACTGCCAGGGATAACCTGGAGGAAGGTGGGGACGACGTCAAATCATCATGCCCCTTATGACCAGGGCTACACACGTGCTACAATGGCGTAAACAGAGGGAAGCAGCCTCGTGAGAGTGAGCAAATCCCAAAAATAACGTCTCAGTTCGGATTGTAGTCTGCAACTCGACTACATGAAGCTGGAATCGCTAGTAATCGCGAATCAGAATGTCGCGGTGAATACGTTCCCGGGTCTTGTACACACCGCCCGTCACACCATGGGAGTCAGTAACGCCCGAAGTCAGTGACCCAACCGTAAGGAGGGAGCTGCCGAAGGCGGGACCGATAACTGGGGTGAAGTCGTAACAAGGTAGCCGTATCGGAAGGTGCGGCTGGATCACCTCCTTT
6                                                                                                                                                                                                                                                                                                                                                                                                                      ATGAAGAGTTTGATCCTGGCTCAGGATGAACGCTAGCTACAGGCTTAACACATGCAAGTCGAGGGGCAGCATGGTCTTAGCTTGCTAAGGCCGATGGCGACCGGCGCACGGGTGAGTAACACGTATCCAACCTGCCGTCTACTCTTGGACAGCCTTCTGAAAGGAAGATTAATACAAGATGGCATCATGAGTCCGCATGTTCACATGATTAAAGGTATTCCGGTAGACGATGGGGATGCGTTCCATTAGATAGTAGGCGGGGTAACGGCCCACCTAGTCTTCGATGGATAGGGGTTCTGAGAGGAAGGTCCCCCACATTGGAACTGAGACACGGTCCAAACTCCTACGGGAGGCAGCAGTGAGGAATATTGGTCAATGGGCGAGAGCCTGAACCAGCCAAGTAGCGTGAAGGATGACTGCCCTATGGGTTGTAAACTTCTTTTATAAAGGAATAAAGTCGGGTATGCATACCCGTTTGCATGTACTTTATGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGAGCGTAGATGGATGTTTAAGTCAGTTGTGAAAGTTTGCGGCTCAACCGTAAAATTGCAGTTGATACTGGATATCTTGAGTGCAGTTGAGGCAGGCGGAATTCGTGGTGTAGCGGTGAAATGCTTAGATATCACGAAGAACTCCGATTGCGAAGGCAGCCTGCTAAGCTGCAACTGACATTGAGGCTCGAAAGTGTGGGTATCAAACAGGATTAGATACCCTGGTAGTCCACACGGTAAACGATGAATACTCGCTGTTTGCGATATACGGCAAGCGGCCAAGCGAAAGCGTTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGAGGAACATGTGGTTTAATTCGATGATACGCGAGGAACCTTACCCGGGCTTAAATTGCAGATGAATTACGGTGAAAGCCGTAAGCCGCAAGGCATCTGTGAAGGTGCTGCATGGTTGTCGTCAGCTCGTGCCGTGAGGTGTCGGCTTAAGTGCCATAACGAGCGCAACCCTTGTTGTCAGTTACTAACA</code></pre>
<pre class="r"><code>write.table(col2_Biobank, file=&#39;input_for_fasta_conversion.txt&#39;, quote=FALSE, sep=&#39;\t&#39;, row.names = F, col.names=F)</code></pre>
</div>
<div id="umap-plotting" class="section level2">
<h2>2. UMAP plotting</h2>
<pre class="r"><code># UMAP plotting
ggplot(df_Biobank, aes(x=umapA1, y=umapA2)) +
  xlab(&#39;\n umap A1&#39;)+
  ylab(&quot;umap A2 \n&quot;) +
  geom_point(aes(fill = genus), shape = 21 ,size=4, alpha=0.9) +
  scale_fill_manual(values = my_color_collection)+
  ggtitle(paste0(&quot;UMAP coordination&quot;,&quot; \n&quot;)) +
  # scale_x_continuous(limits = c(0, max(df_cen_wb$Degree))) +
  # scale_y_continuous(limits = c(0, max(df_cen_wb$Betweenness))) +
  theme(legend.text=element_text(size=13)) + 
  # theme(legend.position=&quot;top&quot;) +
  # theme(legend.title=element_blank()) +
  guides(colour = guide_legend(override.aes = list(size=8), reverse = TRUE))+
  guides(size=FALSE) +
  mytheme_2d</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-3-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_UMAP &lt;- function(df_Biobank){
  ggplot(df_Biobank, aes(x=umapA1, y=umapA2)) +
    xlab(&#39;\n umap A1&#39;)+
    ylab(&quot;umap A2 \n&quot;) +
    geom_point(aes(fill = genus), shape = 21 ,size=4, alpha=0.9) +
    scale_fill_manual(values = my_color_collection)+
    ggtitle(paste0(&quot;UMAP coordination&quot;,&quot; \n&quot;)) +
    # scale_x_continuous(limits = c(0, max(df_cen_wb$Degree))) +
    # scale_y_continuous(limits = c(0, max(df_cen_wb$Betweenness))) +
    theme(legend.text=element_text(size=13)) + 
    # theme(legend.position=&quot;top&quot;) +
    # theme(legend.title=element_blank()) +
    guides(colour = guide_legend(override.aes = list(size=8), reverse = TRUE))+
    guides(size=FALSE) +
    mytheme_2d
}</code></pre>
</div>
<div id="k-means-clustering-for-umap-distribution" class="section level2">
<h2>2.2. K-means clustering for UMAP distribution</h2>
<ol style="list-style-type: decimal">
<li><p>First, using the elbow method to find the optimal number of clusters<br />
Within-Cluster-Sum-of-Squares (WCSS) is the sum of squares of the distances of each data point in all clusters to their respective centroids.<br />
</p></li>
<li></li>
</ol>
<pre class="r"><code># First, using the elbow method to find the optimal number of clusters
set.seed(6)
wcss = vector()

colnames(df_Biobank)</code></pre>
<pre><code> [1] &quot;seq_id&quot;       &quot;umapA1&quot;       &quot;umapA2&quot;       &quot;tiplabel&quot;     &quot;msk_id&quot;      
 [6] &quot;length_bp&quot;    &quot;phylum&quot;       &quot;class&quot;        &quot;order&quot;        &quot;family&quot;      
[11] &quot;genus&quot;        &quot;species&quot;      &quot;nuc_sequence&quot;</code></pre>
<pre class="r"><code>k_dataset &lt;- phylo_Biobank %&gt;% dplyr::select(phy_id, umapA1, umapA2)
k_dataset &lt;- tibble::column_to_rownames(k_dataset, var=&quot;phy_id&quot;)

# 100 clusters
n_clu = 100
for (i in 1:n_clu) wcss[i] = sum(kmeans(k_dataset, i)$withinss)
plot(1:n_clu,
     wcss,
     type = &#39;b&#39;,
     main = paste(&#39;The Elbow Method&#39;),
     xlab = &#39;Number of clusters&#39;,
     ylab = &#39;WCSS&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fitting K-Means to the dataset
set.seed(29)
kmeans = kmeans(x = k_dataset, centers = n_clu)
y_kmeans = kmeans$cluster

# Visualising the clusters
library(cluster)
# All observation are represented by points in the plot, using principal components or multidimensional scaling.
clusplot(k_dataset,
         y_kmeans,
         lines = 0,
         shade = TRUE,
         color = TRUE,
         labels = 0,
         plotchar = FALSE,
         span = TRUE,
         # main = paste(&#39;Clusters of strains with UMAP&#39;),
         main = NA,
         xlab = &#39;PC1&#39;,
         ylab = &#39;PC2&#39;) </code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-4-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># get the centroid and the closest id&#39;s for 100 clusters
kmeans$centers</code></pre>
<pre><code>         umapA1       umapA2
1   -3.46683696   7.90930185
2   -2.84710966   6.20400928
3   -3.02872674   6.22079147
4    1.87679995 -13.61810163
5    0.52403568 -11.29365886
6    1.77458500 -13.58405446
7   -5.51849043  -5.24117348
8   -5.36443173  -5.07259652
9   -1.21108507 -11.30745401
10  -3.22552780   7.36094931
11  -1.68291823 -11.21770212
12   0.62336978 -12.10052317
13  -1.76575698   7.46283564
14   0.11276739 -11.08553461
15  -0.44755765 -13.80478890
16  -3.28938061   5.68213533
17  -3.11014548   5.25269527
18  -0.67669070 -10.64034171
19   5.95334978   1.13854949
20   1.29979249 -11.42110133
21  -0.37598376 -13.73284015
22   2.12073235  10.40458090
23  -4.28350383   5.42166218
24  -3.23387960   5.27487262
25  -2.40425809   5.39707666
26  -2.58222667   5.27718590
27   7.66053407   1.74156998
28  -1.98234434   7.71962290
29  -3.87501532   5.72744032
30   6.18375806   6.62597843
31  -3.26197877   6.75157236
32  -2.97903480   7.03347439
33   5.05393996   6.42153602
34  -5.24522591  -4.94627533
35   6.51284810   1.90201390
36  -1.74636495   8.54755855
37  -1.30855899   8.57420556
38   1.94778996  10.43407628
39   1.81251568 -13.66520024
40  -2.50815126   6.55424999
41  -4.76298507   7.54635294
42  -1.22961313 -11.79310446
43  -3.40329792   6.09919654
44  -0.82681689 -11.36829898
45  -0.17011971 -12.67248016
46   1.55732051   6.93014522
47   0.13617903 -10.91271508
48  -1.48563238   8.79114670
49  -1.85810109   6.65555277
50  -2.97023115   5.62051785
51   6.56417065  -0.03354242
52  -2.46487080   4.86814730
53  -2.71318660   7.40911819
54   0.05040675   8.38770778
55   1.67352639 -11.02851420
56  -3.12874924   5.77911505
57  -4.22785810   5.46359244
58  -0.22609918 -11.68935125
59  -2.61829280   5.48964314
60  -2.05366635   7.30575148
61   1.62826875   6.86197464
62  -2.78157802   8.11919382
63  -3.00588772   6.00256423
64  -1.99708259 -11.00381474
65  -1.02951586 -11.98279124
66   1.50050147 -10.70483182
67   0.63818241 -13.12020841
68  -5.44003558  -5.14262751
69   2.34344044 -11.38848742
70  -1.13114801 -11.79755123
71  -4.46682999   7.38193363
72  -0.40585555 -11.56458157
73  -1.52718275   8.82575595
74  -1.45989873   8.72935790
75  -4.53213396   8.78428815
76  -3.51842218   5.96567695
77  -4.08792857   5.62705706
78   6.97566745   1.39406497
79  -0.31013331 -13.61548366
80  -5.51380908  -5.16740362
81  -3.63757941   7.06861419
82  -2.90118875   7.76979375
83   2.29786646 -10.49222964
84   1.58010475  -9.64660428
85   2.02570661  10.17614630
86  -4.17680229   5.53012718
87  -2.76053082   5.28679817
88   5.74468952   6.70021039
89  -1.40774068   8.63866983
90  -2.46678549   4.97195890
91  -1.36444877   8.74577773
92  -1.51774061   8.60406939
93   1.18609672 -11.41849082
94   1.71354156 -13.49575642
95  -1.03063542 -12.07113871
96  -0.13485841 -12.26774452
97   1.12892203 -10.33602895
98   2.25980085 -10.53172497
99  -2.40835029   7.16321106
100 -4.84593163   7.47841940</code></pre>
<pre class="r"><code>center_strains &lt;- vector()
min_ss &lt;- rep(Inf, n_clu)  # minimum sum of square of distance from centroid

for (i in 1:n_clu){
  print(i)
  centroid_xy &lt;- (kmeans$centers)[i,]
  for (j in 1:dim(k_dataset)[1]){
    point_xy &lt;- k_dataset[j,]
    ss &lt;- sum((centroid_xy - point_xy)^2)
    if (ss &lt; min_ss[i]){
      min_ss[i] &lt;- ss
      center_strains[i] &lt;- rownames(k_dataset[j,])
      # print(center_strains[i])
    }
  }
}</code></pre>
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20
[1] 21
[1] 22
[1] 23
[1] 24
[1] 25
[1] 26
[1] 27
[1] 28
[1] 29
[1] 30
[1] 31
[1] 32
[1] 33
[1] 34
[1] 35
[1] 36
[1] 37
[1] 38
[1] 39
[1] 40
[1] 41
[1] 42
[1] 43
[1] 44
[1] 45
[1] 46
[1] 47
[1] 48
[1] 49
[1] 50
[1] 51
[1] 52
[1] 53
[1] 54
[1] 55
[1] 56
[1] 57
[1] 58
[1] 59
[1] 60
[1] 61
[1] 62
[1] 63
[1] 64
[1] 65
[1] 66
[1] 67
[1] 68
[1] 69
[1] 70
[1] 71
[1] 72
[1] 73
[1] 74
[1] 75
[1] 76
[1] 77
[1] 78
[1] 79
[1] 80
[1] 81
[1] 82
[1] 83
[1] 84
[1] 85
[1] 86
[1] 87
[1] 88
[1] 89
[1] 90
[1] 91
[1] 92
[1] 93
[1] 94
[1] 95
[1] 96
[1] 97
[1] 98
[1] 99
[1] 100</code></pre>
<pre class="r"><code>kmeans$cluster[center_strains] # center strains belong to each cluster</code></pre>
<pre><code>           MSK1_25_Sellimonas_intestinalis 
                                         1 
                        TM121_Blautia_luti 
                                         2 
                        TM139_Blautia_luti 
                                         3 
                 MSK21_60_Prevotella_copri 
                                         4 
  210309-DFI.4.154A_Parabacteroides_merdae 
                                         5 
                 MSK21_56_Prevotella_copri 
                                         6 
              TM406_Escherichia_fergusonii 
                                         7 
                     MSK19_28_unclassified 
                                         8 
               MSK18_85_Bacteroides_caccae 
                                         9 
                   TM448_Dorea_longicatena 
                                        10 
         MSK18_37_Bacteroides_unclassified 
                                        11 
 210120-DFI.1.157_Odoribacter_splanchnicus 
                                        12 
              MSK17_42_Eubacterium_rectale 
                                        13 
210123-DFI.3.77_Parabacteroides_distasonis 
                                        14 
            MSK16_49_Bacteroides_eggerthii 
                                        15 
                       TM175_Blautia_obeum 
                                        16 
                      TM100_Blautia_faecis 
                                        17 
     MSK18_17_Bacteroides_cellulosilyticus 
                                        18 
            FC22_108_Bifidobacterium_breve 
                                        19 
             MSK22_48_Bacteroides_vulgatus 
                                        20 
            MSK16_67_Bacteroides_eggerthii 
                                        21 
   MSK23_76_Erysipelatoclostridium_ramosum 
                                        22 
     TM184_Fusicatenibacter_saccharivorans 
                                        23 
                      TM134_Blautia_faecis 
                                        24 
                    TM104_Blautia_wexlerae 
                                        25 
                 MSK18_55_Blautia_wexlerae 
                                        26 
              TM236_Bifidobacterium_longum 
                                        27 
              MSK22_51_Eubacterium_rectale 
                                        28 
          TM115_Clostridium_celerecrescens 
                                        29 
           MSK2_5_Pediococcus_unclassified 
                                        30 
               TM396_Clostridium_symbosium 
                                        31 
            MSK18_54_Ruminococcus_lactaris 
                                        32 
                FC22_139_Weissella_cibaria 
                                        33 
    EP-10S-DFI-2-07_Escherichia_fergusonii 
                                        34 
                MSK15_90_Eggerthella_lenta 
                                        35 
                 TM460_Anaerostipes_hadrus 
                                        36 
                 TM458_Anaerostipes_hadrus 
                                        37 
      TM554_Erysipelatoclostridium_ramosum 
                                        38 
                 MSK21_74_Prevotella_copri 
                                        39 
   EP-SM-12S-S02_Faecalicatena_fissicatena 
                                        40 
       MSK2_63_Clostridium_clostridioforme 
                                        41 
       MSK5_9_Bacteroides_thetaiotaomicron 
                                        42 
                    TM119_Blautia_producta 
                                        43 
EP-SM-12S-S90_Bacteroides_thetaiotaomicron 
                                        44 
             MSK18_9_Bacteroides_stercoris 
                                        45 
        MSK21_92_Catenibacterium_mitsuokai 
                                        46 
210120-DFI.1.37_Parabacteroides_distasonis 
                                        47 
                 TM485_Anaerostipes_hadrus 
                                        48 
                  MSK5_8_Veillonella_ratti 
                                        49 
                       TM135_Blautia_obeum 
                                        50 
            FC22_097_Bifidobacterium_breve 
                                        51 
                    TM549_Blautia_wexlerae 
                                        52 
                  TM550_Tyzzerella_nexilis 
                                        53 
               TM364_Holdemanella_biformis 
                                        54 
     210125-DFI.1.193_Bacteroides_vulgatus 
                                        55 
                       TM157_Blautia_obeum 
                                        56 
     TM548_Fusicatenibacter_saccharivorans 
                                        57 
                 TM79_Bacteroides_fragilis 
                                        58 
                 MSK19_90_Blautia_wexlerae 
                                        59 
                 TM367_Eubacterium_rectale 
                                        60 
     EP-10S-DFI-2-34_Bacillus_unclassified 
                                        61 
                   TM508_Coprococcus_comes 
                                        62 
                     MSK22_60_Blautia_luti 
                                        63 
        MSK16_41_Bacteroides_xylanisolvens 
                                        64 
     MSK16_60_Bacteroides_thetaiotaomicron 
                                        65 
             MSK20_41_Bacteroides_vulgatus 
                                        66 
          210125-DFI.3.10_Alistipes_shahii 
                                        67 
                     MSK19_17_unclassified 
                                        68 
              MSK17_5_Bacteroides_vulgatus 
                                        69 
        TM551_Bacteroides_thetaiotaomicron 
                                        70 
   210329-SL.2.12_Enterocloster_lavalensis 
                                        71 
                TM350_Bacteroides_fragilis 
                                        72 
                 TM347_Anaerostipes_hadrus 
                                        73 
              MSK22_35_Anaerostipes_hadrus 
                                        74 
               MSK5_19_Ruminococcus_gnavus 
                                        75 
                    TM120_Blautia_producta 
                                        76 
     TM441_Fusicatenibacter_saccharivorans 
                                        77 
        TM393_Bifidobacterium_adolescentis 
                                        78 
            MSK16_66_Bacteroides_eggerthii 
                                        79 
                  MSK18_20_Shigella_sonnei 
                                        80 
                 TM474_Ruminococcus_faecis 
                                        81 
                MSK17_11_Dorea_longicatena 
                                        82 
                TM409_Bacteroides_vulgatus 
                                        83 
             MSK16_47_Bacteroides_vulgatus 
                                        84 
      TM497_Erysipelatoclostridium_ramosum 
                                        85 
   MSK22_1_Fusicatenibacter_saccharivorans 
                                        86 
                     TM91_Blautia_wexlerae 
                                        87 
                   TM47_Lactococcus_lactis 
                                        88 
                 TM377_Anaerostipes_hadrus 
                                        89 
                 MSK22_61_Blautia_wexlerae 
                                        90 
                 TM487_Anaerostipes_hadrus 
                                        91 
                 TM455_Anaerostipes_hadrus 
                                        92 
             MSK22_46_Bacteroides_vulgatus 
                                        93 
                 MSK21_95_Prevotella_copri 
                                        94 
     MSK16_65_Bacteroides_thetaiotaomicron 
                                        95 
            MSK17_15_Bacteroides_uniformis 
                                        96 
             MSK19_20_Bacteroides_vulgatus 
                                        97 
                 TM83_Bacteroides_vulgatus 
                                        98 
       MSK17_41_Intestinibacter_bartlettii 
                                        99 
       MSK2_59_Clostridium_clostridioforme 
                                       100 </code></pre>
<pre class="r"><code>center_strains # confirmed</code></pre>
<pre><code>  [1] &quot;MSK1_25_Sellimonas_intestinalis&quot;           
  [2] &quot;TM121_Blautia_luti&quot;                        
  [3] &quot;TM139_Blautia_luti&quot;                        
  [4] &quot;MSK21_60_Prevotella_copri&quot;                 
  [5] &quot;210309-DFI.4.154A_Parabacteroides_merdae&quot;  
  [6] &quot;MSK21_56_Prevotella_copri&quot;                 
  [7] &quot;TM406_Escherichia_fergusonii&quot;              
  [8] &quot;MSK19_28_unclassified&quot;                     
  [9] &quot;MSK18_85_Bacteroides_caccae&quot;               
 [10] &quot;TM448_Dorea_longicatena&quot;                   
 [11] &quot;MSK18_37_Bacteroides_unclassified&quot;         
 [12] &quot;210120-DFI.1.157_Odoribacter_splanchnicus&quot; 
 [13] &quot;MSK17_42_Eubacterium_rectale&quot;              
 [14] &quot;210123-DFI.3.77_Parabacteroides_distasonis&quot;
 [15] &quot;MSK16_49_Bacteroides_eggerthii&quot;            
 [16] &quot;TM175_Blautia_obeum&quot;                       
 [17] &quot;TM100_Blautia_faecis&quot;                      
 [18] &quot;MSK18_17_Bacteroides_cellulosilyticus&quot;     
 [19] &quot;FC22_108_Bifidobacterium_breve&quot;            
 [20] &quot;MSK22_48_Bacteroides_vulgatus&quot;             
 [21] &quot;MSK16_67_Bacteroides_eggerthii&quot;            
 [22] &quot;MSK23_76_Erysipelatoclostridium_ramosum&quot;   
 [23] &quot;TM184_Fusicatenibacter_saccharivorans&quot;     
 [24] &quot;TM134_Blautia_faecis&quot;                      
 [25] &quot;TM104_Blautia_wexlerae&quot;                    
 [26] &quot;MSK18_55_Blautia_wexlerae&quot;                 
 [27] &quot;TM236_Bifidobacterium_longum&quot;              
 [28] &quot;MSK22_51_Eubacterium_rectale&quot;              
 [29] &quot;TM115_Clostridium_celerecrescens&quot;          
 [30] &quot;MSK2_5_Pediococcus_unclassified&quot;           
 [31] &quot;TM396_Clostridium_symbosium&quot;               
 [32] &quot;MSK18_54_Ruminococcus_lactaris&quot;            
 [33] &quot;FC22_139_Weissella_cibaria&quot;                
 [34] &quot;EP-10S-DFI-2-07_Escherichia_fergusonii&quot;    
 [35] &quot;MSK15_90_Eggerthella_lenta&quot;                
 [36] &quot;TM460_Anaerostipes_hadrus&quot;                 
 [37] &quot;TM458_Anaerostipes_hadrus&quot;                 
 [38] &quot;TM554_Erysipelatoclostridium_ramosum&quot;      
 [39] &quot;MSK21_74_Prevotella_copri&quot;                 
 [40] &quot;EP-SM-12S-S02_Faecalicatena_fissicatena&quot;   
 [41] &quot;MSK2_63_Clostridium_clostridioforme&quot;       
 [42] &quot;MSK5_9_Bacteroides_thetaiotaomicron&quot;       
 [43] &quot;TM119_Blautia_producta&quot;                    
 [44] &quot;EP-SM-12S-S90_Bacteroides_thetaiotaomicron&quot;
 [45] &quot;MSK18_9_Bacteroides_stercoris&quot;             
 [46] &quot;MSK21_92_Catenibacterium_mitsuokai&quot;        
 [47] &quot;210120-DFI.1.37_Parabacteroides_distasonis&quot;
 [48] &quot;TM485_Anaerostipes_hadrus&quot;                 
 [49] &quot;MSK5_8_Veillonella_ratti&quot;                  
 [50] &quot;TM135_Blautia_obeum&quot;                       
 [51] &quot;FC22_097_Bifidobacterium_breve&quot;            
 [52] &quot;TM549_Blautia_wexlerae&quot;                    
 [53] &quot;TM550_Tyzzerella_nexilis&quot;                  
 [54] &quot;TM364_Holdemanella_biformis&quot;               
 [55] &quot;210125-DFI.1.193_Bacteroides_vulgatus&quot;     
 [56] &quot;TM157_Blautia_obeum&quot;                       
 [57] &quot;TM548_Fusicatenibacter_saccharivorans&quot;     
 [58] &quot;TM79_Bacteroides_fragilis&quot;                 
 [59] &quot;MSK19_90_Blautia_wexlerae&quot;                 
 [60] &quot;TM367_Eubacterium_rectale&quot;                 
 [61] &quot;EP-10S-DFI-2-34_Bacillus_unclassified&quot;     
 [62] &quot;TM508_Coprococcus_comes&quot;                   
 [63] &quot;MSK22_60_Blautia_luti&quot;                     
 [64] &quot;MSK16_41_Bacteroides_xylanisolvens&quot;        
 [65] &quot;MSK16_60_Bacteroides_thetaiotaomicron&quot;     
 [66] &quot;MSK20_41_Bacteroides_vulgatus&quot;             
 [67] &quot;210125-DFI.3.10_Alistipes_shahii&quot;          
 [68] &quot;MSK19_17_unclassified&quot;                     
 [69] &quot;MSK17_5_Bacteroides_vulgatus&quot;              
 [70] &quot;TM551_Bacteroides_thetaiotaomicron&quot;        
 [71] &quot;210329-SL.2.12_Enterocloster_lavalensis&quot;   
 [72] &quot;TM350_Bacteroides_fragilis&quot;                
 [73] &quot;TM347_Anaerostipes_hadrus&quot;                 
 [74] &quot;MSK22_35_Anaerostipes_hadrus&quot;              
 [75] &quot;MSK5_19_Ruminococcus_gnavus&quot;               
 [76] &quot;TM120_Blautia_producta&quot;                    
 [77] &quot;TM441_Fusicatenibacter_saccharivorans&quot;     
 [78] &quot;TM393_Bifidobacterium_adolescentis&quot;        
 [79] &quot;MSK16_66_Bacteroides_eggerthii&quot;            
 [80] &quot;MSK18_20_Shigella_sonnei&quot;                  
 [81] &quot;TM474_Ruminococcus_faecis&quot;                 
 [82] &quot;MSK17_11_Dorea_longicatena&quot;                
 [83] &quot;TM409_Bacteroides_vulgatus&quot;                
 [84] &quot;MSK16_47_Bacteroides_vulgatus&quot;             
 [85] &quot;TM497_Erysipelatoclostridium_ramosum&quot;      
 [86] &quot;MSK22_1_Fusicatenibacter_saccharivorans&quot;   
 [87] &quot;TM91_Blautia_wexlerae&quot;                     
 [88] &quot;TM47_Lactococcus_lactis&quot;                   
 [89] &quot;TM377_Anaerostipes_hadrus&quot;                 
 [90] &quot;MSK22_61_Blautia_wexlerae&quot;                 
 [91] &quot;TM487_Anaerostipes_hadrus&quot;                 
 [92] &quot;TM455_Anaerostipes_hadrus&quot;                 
 [93] &quot;MSK22_46_Bacteroides_vulgatus&quot;             
 [94] &quot;MSK21_95_Prevotella_copri&quot;                 
 [95] &quot;MSK16_65_Bacteroides_thetaiotaomicron&quot;     
 [96] &quot;MSK17_15_Bacteroides_uniformis&quot;            
 [97] &quot;MSK19_20_Bacteroides_vulgatus&quot;             
 [98] &quot;TM83_Bacteroides_vulgatus&quot;                 
 [99] &quot;MSK17_41_Intestinibacter_bartlettii&quot;       
[100] &quot;MSK2_59_Clostridium_clostridioforme&quot;       </code></pre>
<pre class="r"><code>length(center_strains)</code></pre>
<pre><code>[1] 100</code></pre>
<pre class="r"><code># Plot UMAP with cluster center strains
plot_UMAP(phylo_Biobank %&gt;% filter(phy_id %in% center_strains))</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-4-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># number of species
df_center_umap &lt;- phylo_Biobank %&gt;% filter(phy_id %in% center_strains)
df_center_umap %&gt;% dplyr::select(species) %&gt;% unique() %&gt;% dim() # 51 species</code></pre>
<pre><code>[1] 51  1</code></pre>
<pre class="r"><code>df_umap_distinct &lt;- df_center_umap %&gt;% count(species)
hist(df_umap_distinct$n, main=&quot;Histogram of number of same species&quot;, xlab=&quot;Number of same species&quot;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-4-4.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Try it again with 20 clusters</p>
<pre class="r"><code># 20 clusters
set.seed(6)
wcss = vector()
n_clu = 20
for (i in 1:n_clu) wcss[i] = sum(kmeans(k_dataset, i)$withinss)
plot(1:n_clu,
     wcss,
     type = &#39;b&#39;,
     main = paste(&#39;The Elbow Method&#39;),
     xlab = &#39;Number of clusters&#39;,
     ylab = &#39;WCSS&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-5-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fitting K-Means to the dataset
set.seed(29)
kmeans = kmeans(x = k_dataset, centers = n_clu)
y_kmeans = kmeans$cluster

# Visualising the clusters
library(cluster)
# All observation are represented by points in the plot, using principal components or multidimensional scaling.
clusplot(k_dataset,
         y_kmeans,
         lines = 0,
         shade = TRUE,
         color = TRUE,
         labels = 0,
         plotchar = FALSE,
         span = TRUE,
         # main = paste(&#39;Clusters of strains with UMAP&#39;),
         main = NA,
         xlab = &#39;PC1&#39;,
         ylab = &#39;PC2&#39;) </code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-5-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># get the centroid and the closest id&#39;s for 100 clusters
kmeans$centers</code></pre>
<pre><code>       umapA1     umapA2
1  -4.5741750   8.423395
2  -0.9534972   8.596058
3  -2.6163340   6.482426
4   1.8235875 -13.619438
5   0.2452385 -11.949624
6   0.5703019 -13.067964
7  -5.4683288  -5.158901
8  -5.2661412  -4.969316
9  -1.0062511 -11.607182
10 -2.7443975   7.557665
11 -1.8242922 -11.121453
12 -0.1898748 -12.346391
13  1.9464752   9.181541
14  0.1848981 -11.016513
15 -0.3812795 -13.722815
16 -3.8704702   5.702786
17 -2.7258208   5.337540
18 -0.6766907 -10.640342
19  6.6397061   2.866920
20  1.5813373 -10.720570</code></pre>
<pre class="r"><code>center_strains &lt;- vector()
min_ss &lt;- rep(Inf, n_clu)  # minimum sum of square of distance from centroid

for (i in 1:n_clu){
  print(i)
  centroid_xy &lt;- (kmeans$centers)[i,]
  for (j in 1:dim(k_dataset)[1]){
    point_xy &lt;- k_dataset[j,]
    ss &lt;- sum((centroid_xy - point_xy)^2)
    if (ss &lt; min_ss[i]){
      min_ss[i] &lt;- ss
      center_strains[i] &lt;- rownames(k_dataset[j,])
      # print(center_strains[i])
    }
  }
}</code></pre>
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20</code></pre>
<pre class="r"><code>kmeans$cluster[center_strains] # center strains belong to each cluster</code></pre>
<pre><code>                 MSK22_53_Ruminococcus_gnavus 
                                            1 
                    TM456_Anaerostipes_hadrus 
                                            2 
              TM419_Faecalicatena_fissicatena 
                                            3 
                    MSK21_37_Prevotella_copri 
                                            4 
               MSK20_79_Bacteroides_uniformis 
                                            5 
           EP-SM-12S-S89_Alistipes_finegoldii 
                                            6 
                      MSK5_13_Shigella_sonnei 
                                            7 
       EP-10S-DFI-2-39_Escherichia_fergusonii 
                                            8 
210123-DFI.4.133_Bacteroides_thetaiotaomicron 
                                            9 
               MSK17_61_Dorea_formicigenerans 
                                           10 
                     TM544_Bacteroides_ovatus 
                                           11 
               MSK18_74_Bacteroides_uniformis 
                                           12 
         TM501_Erysipelatoclostridium_ramosum 
                                           13 
          MSK22_22_Parabacteroides_distasonis 
                                           14 
               MSK16_67_Bacteroides_eggerthii 
                                           15 
     MSK22_67_Fusicatenibacter_saccharivorans 
                                           16 
                    MSK15_46_Blautia_wexlerae 
                                           17 
        MSK18_17_Bacteroides_cellulosilyticus 
                                           18 
                TM76_Collinsella_unclassified 
                                           19 
                MSK20_41_Bacteroides_vulgatus 
                                           20 </code></pre>
<pre class="r"><code>center_strains # confirmed</code></pre>
<pre><code> [1] &quot;MSK22_53_Ruminococcus_gnavus&quot;                 
 [2] &quot;TM456_Anaerostipes_hadrus&quot;                    
 [3] &quot;TM419_Faecalicatena_fissicatena&quot;              
 [4] &quot;MSK21_37_Prevotella_copri&quot;                    
 [5] &quot;MSK20_79_Bacteroides_uniformis&quot;               
 [6] &quot;EP-SM-12S-S89_Alistipes_finegoldii&quot;           
 [7] &quot;MSK5_13_Shigella_sonnei&quot;                      
 [8] &quot;EP-10S-DFI-2-39_Escherichia_fergusonii&quot;       
 [9] &quot;210123-DFI.4.133_Bacteroides_thetaiotaomicron&quot;
[10] &quot;MSK17_61_Dorea_formicigenerans&quot;               
[11] &quot;TM544_Bacteroides_ovatus&quot;                     
[12] &quot;MSK18_74_Bacteroides_uniformis&quot;               
[13] &quot;TM501_Erysipelatoclostridium_ramosum&quot;         
[14] &quot;MSK22_22_Parabacteroides_distasonis&quot;          
[15] &quot;MSK16_67_Bacteroides_eggerthii&quot;               
[16] &quot;MSK22_67_Fusicatenibacter_saccharivorans&quot;     
[17] &quot;MSK15_46_Blautia_wexlerae&quot;                    
[18] &quot;MSK18_17_Bacteroides_cellulosilyticus&quot;        
[19] &quot;TM76_Collinsella_unclassified&quot;                
[20] &quot;MSK20_41_Bacteroides_vulgatus&quot;                </code></pre>
<pre class="r"><code>length(center_strains)</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code># Plot UMAP with cluster center strains
plot_UMAP(phylo_Biobank %&gt;% filter(phy_id %in% center_strains))</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-5-3.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="phylogenetic-tree-construction-and-cut-the-tree-clustering" class="section level2">
<h2>3. Phylogenetic tree construction and cut the tree / clustering</h2>
<pre class="r"><code>phy_tree &lt;- ape::read.tree(&quot;data/Biobank_align_mafft.fasta.treefile_rooted.nwk&quot;)
phy_tree</code></pre>
<pre><code>
Phylogenetic tree with 850 tips and 849 internal nodes.

Tip labels:
  210119-DFI.3.32_Escherichia_fergusonii, EP-10S-DFI-2-08_Escherichia_fergusonii, MSK5_13_Shigella_sonnei, MSK17_69_Escherichia_fergusonii, 210309-SL2.01_Escherichia_fergusonii, MSK14_41_Escherichia_fergusonii, ...

Rooted; includes branch lengths.</code></pre>
<pre class="r"><code># will be using cutree function in hclust package. In order to do so, we need to make it ultrametric
# not ultrametric
is.ultrametric(phy_tree)</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># convert to ultrametric tree 
# (http://blog.phytools.org/2017/03/forceultrametric-method-for-ultrametric.html)
force.ultrametric&lt;-function(tree,method=c(&quot;nnls&quot;,&quot;extend&quot;)){
    method&lt;-method[1]
    if(method==&quot;nnls&quot;) tree&lt;-nnls.tree(cophenetic(tree),tree,
        rooted=TRUE,trace=0)
    else if(method==&quot;extend&quot;){
        h&lt;-diag(vcv(tree))
        d&lt;-max(h)-h
        ii&lt;-sapply(1:Ntip(tree),function(x,y) which(y==x),
            y=tree$edge[,2])
        tree$edge.length[ii]&lt;-tree$edge.length[ii]+d
    } else 
        cat(&quot;method not recognized: returning input tree\n\n&quot;)
    tree
}

library(phangorn)
ult_tree &lt;- force.ultrametric(phy_tree)
is.ultrametric(ult_tree) # true</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># convert to hclust object
hcl &lt;- as.hclust(ult_tree)
plot(hcl,labels = FALSE, hang= -1)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-6-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>The cutree() function provides the functionality to output either desired number of clusters or clusters obtained from cutting the dendrogram at a certain height.</p>
<pre class="r"><code># cut tree
cut_tree &lt;- cutree(hcl, k=100)
table(cut_tree) </code></pre>
<pre><code>cut_tree
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
  1   1   3  17   3  29   2   3  40   7   1   7   9   1   4   1   3   8   1   1 
 21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40 
  4  13  18   1   7  23   1   1   1   5  22  24   3   1   8   4   2  35   1   5 
 41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60 
  1   4   3   1   6   8  14  33   1   1   1   1   2  10   2   1  25   2   1   1 
 61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80 
  1   3   2   2   1   4   6   1   1   1   1  72   2   3  12   7   4   5  32  29 
 81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 
  1   7   8  31  20   5  88   1   5  18   2   3  11   4   9   1   1   5   1   1 </code></pre>
<pre class="r"><code>hist(as.vector(table(cut_tree)), breaks=100)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-7-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># table
table(cut_tree)</code></pre>
<pre><code>cut_tree
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
  1   1   3  17   3  29   2   3  40   7   1   7   9   1   4   1   3   8   1   1 
 21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40 
  4  13  18   1   7  23   1   1   1   5  22  24   3   1   8   4   2  35   1   5 
 41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60 
  1   4   3   1   6   8  14  33   1   1   1   1   2  10   2   1  25   2   1   1 
 61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80 
  1   3   2   2   1   4   6   1   1   1   1  72   2   3  12   7   4   5  32  29 
 81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 
  1   7   8  31  20   5  88   1   5  18   2   3  11   4   9   1   1   5   1   1 </code></pre>
<pre class="r"><code>head(cut_tree)</code></pre>
<pre><code>210119-DFI.3.32_Escherichia_fergusonii EP-10S-DFI-2-08_Escherichia_fergusonii 
                                     1                                      2 
               MSK5_13_Shigella_sonnei        MSK17_69_Escherichia_fergusonii 
                                     3                                      3 
  210309-SL2.01_Escherichia_fergusonii        MSK14_41_Escherichia_fergusonii 
                                     3                                      4 </code></pre>
<pre class="r"><code># this is not a good method to cluster</code></pre>
</div>
<div id="use-distance-matrix-to-do-k-means-clustering" class="section level2">
<h2>3.2. Use distance matrix to do k-means clustering</h2>
<pre class="r"><code># input tab delimited table
ml_dist &lt;- read.table(&quot;data/Biobank_align_mafft.fasta.mldist.tsv&quot;, sep=&quot;\t&quot;, header = F)
# head(ml_dist)

# set row and column name
ml_dist %&lt;&gt;% dplyr::select(-V531)
dim(ml_dist)</code></pre>
<pre><code>[1] 529 530</code></pre>
<pre class="r"><code>colnames(ml_dist)</code></pre>
<pre><code>  [1] &quot;V1&quot;   &quot;V2&quot;   &quot;V3&quot;   &quot;V4&quot;   &quot;V5&quot;   &quot;V6&quot;   &quot;V7&quot;   &quot;V8&quot;   &quot;V9&quot;   &quot;V10&quot; 
 [11] &quot;V11&quot;  &quot;V12&quot;  &quot;V13&quot;  &quot;V14&quot;  &quot;V15&quot;  &quot;V16&quot;  &quot;V17&quot;  &quot;V18&quot;  &quot;V19&quot;  &quot;V20&quot; 
 [21] &quot;V21&quot;  &quot;V22&quot;  &quot;V23&quot;  &quot;V24&quot;  &quot;V25&quot;  &quot;V26&quot;  &quot;V27&quot;  &quot;V28&quot;  &quot;V29&quot;  &quot;V30&quot; 
 [31] &quot;V31&quot;  &quot;V32&quot;  &quot;V33&quot;  &quot;V34&quot;  &quot;V35&quot;  &quot;V36&quot;  &quot;V37&quot;  &quot;V38&quot;  &quot;V39&quot;  &quot;V40&quot; 
 [41] &quot;V41&quot;  &quot;V42&quot;  &quot;V43&quot;  &quot;V44&quot;  &quot;V45&quot;  &quot;V46&quot;  &quot;V47&quot;  &quot;V48&quot;  &quot;V49&quot;  &quot;V50&quot; 
 [51] &quot;V51&quot;  &quot;V52&quot;  &quot;V53&quot;  &quot;V54&quot;  &quot;V55&quot;  &quot;V56&quot;  &quot;V57&quot;  &quot;V58&quot;  &quot;V59&quot;  &quot;V60&quot; 
 [61] &quot;V61&quot;  &quot;V62&quot;  &quot;V63&quot;  &quot;V64&quot;  &quot;V65&quot;  &quot;V66&quot;  &quot;V67&quot;  &quot;V68&quot;  &quot;V69&quot;  &quot;V70&quot; 
 [71] &quot;V71&quot;  &quot;V72&quot;  &quot;V73&quot;  &quot;V74&quot;  &quot;V75&quot;  &quot;V76&quot;  &quot;V77&quot;  &quot;V78&quot;  &quot;V79&quot;  &quot;V80&quot; 
 [81] &quot;V81&quot;  &quot;V82&quot;  &quot;V83&quot;  &quot;V84&quot;  &quot;V85&quot;  &quot;V86&quot;  &quot;V87&quot;  &quot;V88&quot;  &quot;V89&quot;  &quot;V90&quot; 
 [91] &quot;V91&quot;  &quot;V92&quot;  &quot;V93&quot;  &quot;V94&quot;  &quot;V95&quot;  &quot;V96&quot;  &quot;V97&quot;  &quot;V98&quot;  &quot;V99&quot;  &quot;V100&quot;
[101] &quot;V101&quot; &quot;V102&quot; &quot;V103&quot; &quot;V104&quot; &quot;V105&quot; &quot;V106&quot; &quot;V107&quot; &quot;V108&quot; &quot;V109&quot; &quot;V110&quot;
[111] &quot;V111&quot; &quot;V112&quot; &quot;V113&quot; &quot;V114&quot; &quot;V115&quot; &quot;V116&quot; &quot;V117&quot; &quot;V118&quot; &quot;V119&quot; &quot;V120&quot;
[121] &quot;V121&quot; &quot;V122&quot; &quot;V123&quot; &quot;V124&quot; &quot;V125&quot; &quot;V126&quot; &quot;V127&quot; &quot;V128&quot; &quot;V129&quot; &quot;V130&quot;
[131] &quot;V131&quot; &quot;V132&quot; &quot;V133&quot; &quot;V134&quot; &quot;V135&quot; &quot;V136&quot; &quot;V137&quot; &quot;V138&quot; &quot;V139&quot; &quot;V140&quot;
[141] &quot;V141&quot; &quot;V142&quot; &quot;V143&quot; &quot;V144&quot; &quot;V145&quot; &quot;V146&quot; &quot;V147&quot; &quot;V148&quot; &quot;V149&quot; &quot;V150&quot;
[151] &quot;V151&quot; &quot;V152&quot; &quot;V153&quot; &quot;V154&quot; &quot;V155&quot; &quot;V156&quot; &quot;V157&quot; &quot;V158&quot; &quot;V159&quot; &quot;V160&quot;
[161] &quot;V161&quot; &quot;V162&quot; &quot;V163&quot; &quot;V164&quot; &quot;V165&quot; &quot;V166&quot; &quot;V167&quot; &quot;V168&quot; &quot;V169&quot; &quot;V170&quot;
[171] &quot;V171&quot; &quot;V172&quot; &quot;V173&quot; &quot;V174&quot; &quot;V175&quot; &quot;V176&quot; &quot;V177&quot; &quot;V178&quot; &quot;V179&quot; &quot;V180&quot;
[181] &quot;V181&quot; &quot;V182&quot; &quot;V183&quot; &quot;V184&quot; &quot;V185&quot; &quot;V186&quot; &quot;V187&quot; &quot;V188&quot; &quot;V189&quot; &quot;V190&quot;
[191] &quot;V191&quot; &quot;V192&quot; &quot;V193&quot; &quot;V194&quot; &quot;V195&quot; &quot;V196&quot; &quot;V197&quot; &quot;V198&quot; &quot;V199&quot; &quot;V200&quot;
[201] &quot;V201&quot; &quot;V202&quot; &quot;V203&quot; &quot;V204&quot; &quot;V205&quot; &quot;V206&quot; &quot;V207&quot; &quot;V208&quot; &quot;V209&quot; &quot;V210&quot;
[211] &quot;V211&quot; &quot;V212&quot; &quot;V213&quot; &quot;V214&quot; &quot;V215&quot; &quot;V216&quot; &quot;V217&quot; &quot;V218&quot; &quot;V219&quot; &quot;V220&quot;
[221] &quot;V221&quot; &quot;V222&quot; &quot;V223&quot; &quot;V224&quot; &quot;V225&quot; &quot;V226&quot; &quot;V227&quot; &quot;V228&quot; &quot;V229&quot; &quot;V230&quot;
[231] &quot;V231&quot; &quot;V232&quot; &quot;V233&quot; &quot;V234&quot; &quot;V235&quot; &quot;V236&quot; &quot;V237&quot; &quot;V238&quot; &quot;V239&quot; &quot;V240&quot;
[241] &quot;V241&quot; &quot;V242&quot; &quot;V243&quot; &quot;V244&quot; &quot;V245&quot; &quot;V246&quot; &quot;V247&quot; &quot;V248&quot; &quot;V249&quot; &quot;V250&quot;
[251] &quot;V251&quot; &quot;V252&quot; &quot;V253&quot; &quot;V254&quot; &quot;V255&quot; &quot;V256&quot; &quot;V257&quot; &quot;V258&quot; &quot;V259&quot; &quot;V260&quot;
[261] &quot;V261&quot; &quot;V262&quot; &quot;V263&quot; &quot;V264&quot; &quot;V265&quot; &quot;V266&quot; &quot;V267&quot; &quot;V268&quot; &quot;V269&quot; &quot;V270&quot;
[271] &quot;V271&quot; &quot;V272&quot; &quot;V273&quot; &quot;V274&quot; &quot;V275&quot; &quot;V276&quot; &quot;V277&quot; &quot;V278&quot; &quot;V279&quot; &quot;V280&quot;
[281] &quot;V281&quot; &quot;V282&quot; &quot;V283&quot; &quot;V284&quot; &quot;V285&quot; &quot;V286&quot; &quot;V287&quot; &quot;V288&quot; &quot;V289&quot; &quot;V290&quot;
[291] &quot;V291&quot; &quot;V292&quot; &quot;V293&quot; &quot;V294&quot; &quot;V295&quot; &quot;V296&quot; &quot;V297&quot; &quot;V298&quot; &quot;V299&quot; &quot;V300&quot;
[301] &quot;V301&quot; &quot;V302&quot; &quot;V303&quot; &quot;V304&quot; &quot;V305&quot; &quot;V306&quot; &quot;V307&quot; &quot;V308&quot; &quot;V309&quot; &quot;V310&quot;
[311] &quot;V311&quot; &quot;V312&quot; &quot;V313&quot; &quot;V314&quot; &quot;V315&quot; &quot;V316&quot; &quot;V317&quot; &quot;V318&quot; &quot;V319&quot; &quot;V320&quot;
[321] &quot;V321&quot; &quot;V322&quot; &quot;V323&quot; &quot;V324&quot; &quot;V325&quot; &quot;V326&quot; &quot;V327&quot; &quot;V328&quot; &quot;V329&quot; &quot;V330&quot;
[331] &quot;V331&quot; &quot;V332&quot; &quot;V333&quot; &quot;V334&quot; &quot;V335&quot; &quot;V336&quot; &quot;V337&quot; &quot;V338&quot; &quot;V339&quot; &quot;V340&quot;
[341] &quot;V341&quot; &quot;V342&quot; &quot;V343&quot; &quot;V344&quot; &quot;V345&quot; &quot;V346&quot; &quot;V347&quot; &quot;V348&quot; &quot;V349&quot; &quot;V350&quot;
[351] &quot;V351&quot; &quot;V352&quot; &quot;V353&quot; &quot;V354&quot; &quot;V355&quot; &quot;V356&quot; &quot;V357&quot; &quot;V358&quot; &quot;V359&quot; &quot;V360&quot;
[361] &quot;V361&quot; &quot;V362&quot; &quot;V363&quot; &quot;V364&quot; &quot;V365&quot; &quot;V366&quot; &quot;V367&quot; &quot;V368&quot; &quot;V369&quot; &quot;V370&quot;
[371] &quot;V371&quot; &quot;V372&quot; &quot;V373&quot; &quot;V374&quot; &quot;V375&quot; &quot;V376&quot; &quot;V377&quot; &quot;V378&quot; &quot;V379&quot; &quot;V380&quot;
[381] &quot;V381&quot; &quot;V382&quot; &quot;V383&quot; &quot;V384&quot; &quot;V385&quot; &quot;V386&quot; &quot;V387&quot; &quot;V388&quot; &quot;V389&quot; &quot;V390&quot;
[391] &quot;V391&quot; &quot;V392&quot; &quot;V393&quot; &quot;V394&quot; &quot;V395&quot; &quot;V396&quot; &quot;V397&quot; &quot;V398&quot; &quot;V399&quot; &quot;V400&quot;
[401] &quot;V401&quot; &quot;V402&quot; &quot;V403&quot; &quot;V404&quot; &quot;V405&quot; &quot;V406&quot; &quot;V407&quot; &quot;V408&quot; &quot;V409&quot; &quot;V410&quot;
[411] &quot;V411&quot; &quot;V412&quot; &quot;V413&quot; &quot;V414&quot; &quot;V415&quot; &quot;V416&quot; &quot;V417&quot; &quot;V418&quot; &quot;V419&quot; &quot;V420&quot;
[421] &quot;V421&quot; &quot;V422&quot; &quot;V423&quot; &quot;V424&quot; &quot;V425&quot; &quot;V426&quot; &quot;V427&quot; &quot;V428&quot; &quot;V429&quot; &quot;V430&quot;
[431] &quot;V431&quot; &quot;V432&quot; &quot;V433&quot; &quot;V434&quot; &quot;V435&quot; &quot;V436&quot; &quot;V437&quot; &quot;V438&quot; &quot;V439&quot; &quot;V440&quot;
[441] &quot;V441&quot; &quot;V442&quot; &quot;V443&quot; &quot;V444&quot; &quot;V445&quot; &quot;V446&quot; &quot;V447&quot; &quot;V448&quot; &quot;V449&quot; &quot;V450&quot;
[451] &quot;V451&quot; &quot;V452&quot; &quot;V453&quot; &quot;V454&quot; &quot;V455&quot; &quot;V456&quot; &quot;V457&quot; &quot;V458&quot; &quot;V459&quot; &quot;V460&quot;
[461] &quot;V461&quot; &quot;V462&quot; &quot;V463&quot; &quot;V464&quot; &quot;V465&quot; &quot;V466&quot; &quot;V467&quot; &quot;V468&quot; &quot;V469&quot; &quot;V470&quot;
[471] &quot;V471&quot; &quot;V472&quot; &quot;V473&quot; &quot;V474&quot; &quot;V475&quot; &quot;V476&quot; &quot;V477&quot; &quot;V478&quot; &quot;V479&quot; &quot;V480&quot;
[481] &quot;V481&quot; &quot;V482&quot; &quot;V483&quot; &quot;V484&quot; &quot;V485&quot; &quot;V486&quot; &quot;V487&quot; &quot;V488&quot; &quot;V489&quot; &quot;V490&quot;
[491] &quot;V491&quot; &quot;V492&quot; &quot;V493&quot; &quot;V494&quot; &quot;V495&quot; &quot;V496&quot; &quot;V497&quot; &quot;V498&quot; &quot;V499&quot; &quot;V500&quot;
[501] &quot;V501&quot; &quot;V502&quot; &quot;V503&quot; &quot;V504&quot; &quot;V505&quot; &quot;V506&quot; &quot;V507&quot; &quot;V508&quot; &quot;V509&quot; &quot;V510&quot;
[511] &quot;V511&quot; &quot;V512&quot; &quot;V513&quot; &quot;V514&quot; &quot;V515&quot; &quot;V516&quot; &quot;V517&quot; &quot;V518&quot; &quot;V519&quot; &quot;V520&quot;
[521] &quot;V521&quot; &quot;V522&quot; &quot;V523&quot; &quot;V524&quot; &quot;V525&quot; &quot;V526&quot; &quot;V527&quot; &quot;V528&quot; &quot;V529&quot; &quot;V530&quot;</code></pre>
<pre class="r"><code>tibble::column_to_rownames(ml_dist, var=&quot;V1&quot;) %&gt;% dim()</code></pre>
<pre><code>[1] 529 529</code></pre>
<pre class="r"><code>mat_dist &lt;- tibble::column_to_rownames(ml_dist, var=&quot;V1&quot;)
colnames(mat_dist) &lt;- rownames(mat_dist)
dim(mat_dist)</code></pre>
<pre><code>[1] 529 529</code></pre>
<pre class="r"><code>mat_dist &lt;- as.matrix(mat_dist)</code></pre>
<p>K-means clustering</p>
<pre class="r"><code># distance matrix has 529 x 529 dimension
library(FCPS)

kmeans_100 &lt;- kmeansClustering(mat_dist, ClusterNo=100, RandomNo=1, PlotIt= T, Verbose = T)</code></pre>
<pre><code>[1] &quot;Starting iteration 1&quot;
[1] &quot;Starting iteration 2&quot;
[1] &quot;Starting iteration 3&quot;
[1] &quot;Starting iteration 4&quot;
[1] &quot;Starting iteration 5&quot;
[1] &quot;Starting iteration 6&quot;</code></pre>
<pre class="r"><code>kmeans_cls &lt;- kmeans_100$Cls
table(kmeans_cls)</code></pre>
<pre><code>kmeans_cls
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
 13  16   4   2  16   2   7   1   5   2   6   1   1  16  14  21  15   1   8   2 
 21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40 
  2   6   5   4   3   4   2  18   1  15   3  18   7   1   6   8   1   2   3   2 
 41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60 
  9   1   9   4   2   2   2   3  19   1   2   1   5   1   1   8   3   3   1   3 
 61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80 
 27   1   1  10   4   2   5   1   6   1   3   2  10   6   2   4   1   6   1   1 
 81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 
 13   1   1   2   2   3   1  14   2  16   1   2   2   1   2   1   1   1  24   1 </code></pre>
<pre class="r"><code># Histogram of # of strains belonging to each cluster
hist(as.vector(table(kmeans_cls)), breaks=100)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-9-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Let&#39;s get the list of center ids
# kmeans_100

kmeans_100$centerids</code></pre>
<pre><code>  [1] 199 295 442 449 229 516 378 225 281 119 147 109 448 139 403  82 485  75
 [19] 211 426 446 105 454 324 435 438 495 351 107 501 528  37 521 320 493 175
 [37] 428 332 491 431 410 336 166 184 186 429 433 450 312 215 387 386 262 323
 [55]  30 272 373  35 161 330  49 165 160 363  56 321 131 170 171 202 244 157
 [73] 339 266 327 251 158 393 155 159 484 525 197 459 334 423 527 121 388  91
 [91] 319 400 478 318 209 280 250 127  21 148</code></pre>
<pre class="r"><code>phy_id_list &lt;- names(kmeans_cls[kmeans_100$centerids])

# function to get the centroid ids
get_centroids &lt;- function(mat_dist, num_cluster=100){
  kmeans_100 &lt;- kmeansClustering(mat_dist, ClusterNo=num_cluster, RandomNo=1, PlotIt= T, Verbose = T)
  kmeans_cls &lt;- kmeans_100$Cls
  table(kmeans_cls)
  
  # Let&#39;s get the list of center ids
  phy_id_list &lt;- names(kmeans_cls[kmeans_100$centerids])
  return(phy_id_list)
}

# get 100 centroids and plot UMAP
phy_id_list &lt;- get_centroids(mat_dist, num_cluster=100)</code></pre>
<pre><code>[1] &quot;Starting iteration 1&quot;
[1] &quot;Starting iteration 2&quot;
[1] &quot;Starting iteration 3&quot;
[1] &quot;Starting iteration 4&quot;
[1] &quot;Starting iteration 5&quot;</code></pre>
<pre class="r"><code>df_centroid_100 &lt;- phylo_Biobank %&gt;% filter(phy_id %in% phy_id_list)

plot_UMAP(phylo_Biobank %&gt;% filter(phy_id %in% phy_id_list))</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-9-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># number of species
df_center_phy &lt;- phylo_Biobank %&gt;% filter(phy_id %in% phy_id_list)
df_center_phy %&gt;% dplyr::select(species) %&gt;% unique() %&gt;% dim() # 47 species</code></pre>
<pre><code>[1] 60  1</code></pre>
<pre class="r"><code>df_phy_distinct &lt;- df_center_phy %&gt;% count(species)
hist(df_phy_distinct$n, main=&quot;Histogram of number of same species&quot;, xlab=&quot;Number of same species&quot;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-9-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># See what species the centerids are
phylo_Biobank %&gt;% filter(phy_id %in% phy_id_list) %&gt;% dplyr::select(phy_id, species) %&gt;% arrange(species)</code></pre>
<pre><code>                                                 phy_id
1                  MSK17_8_Acidaminococcus_unclassified
2                  210215-DFI.2.31_Alistipes_finegoldii
3                          MSK15_34_Anaerostipes_hadrus
4                          MSK22_35_Anaerostipes_hadrus
5                             TM440_Anaerostipes_hadrus
6                             TM443_Anaerostipes_hadrus
7                             TM447_Anaerostipes_hadrus
8                             TM458_Anaerostipes_hadrus
9                 EP-10S-DFI-2-34_Bacillus_unclassified
10                   210119-DFI.4.132_Bacteroides_dorei
11                           TM242_Bacteroides_fragilis
12                    MSK20_40_Bacteroides_massiliensis
13               210123-DFI.4.65_Bacteroides_salyersiae
14                       MSK18_62_Bacteroides_stercoris
15         210215-DFI.2.69_Bacteroides_thetaiotaomicron
16                 MSK18_2_Bacteroides_thetaiotaomicron
17                 MSK5_14_Bacteroides_thetaiotaomicron
18                  MSK5_5_Bacteroides_thetaiotaomicron
19                    MSK18_22_Bacteroides_unclassified
20                    MSK18_39_Bacteroides_unclassified
21                       MSK18_30_Bacteroides_uniformis
22                        MSK22_36_Bacteroides_vulgatus
23                 MSK20_2_Bifidobacterium_adolescentis
24                 MSK5_20_Bifidobacterium_adolescentis
25                   TM240_Bifidobacterium_adolescentis
26                   TM541_Bifidobacterium_adolescentis
27                      MSK19_51_Bifidobacterium_longum
28                   TM341_Bifidobacterium_unclassified
29                                 TM100_Blautia_faecis
30                               TM141_Blautia_hansenii
31                                   TM121_Blautia_luti
32                                    TM93_Blautia_luti
33                                  TM174_Blautia_obeum
34                                   TM98_Blautia_obeum
35                               TM118_Blautia_producta
36                               TM122_Blautia_schinkii
37                            MSK15_46_Blautia_wexlerae
38                            MSK18_16_Blautia_wexlerae
39                            MSK18_55_Blautia_wexlerae
40                            MSK23_75_Blautia_wexlerae
41                               TM102_Blautia_wexlerae
42                                TM91_Blautia_wexlerae
43                   MSK21_33_Catenibacterium_mitsuokai
44                        MSK1_13_Clostridium_aldenense
45                  MSK2_63_Clostridium_clostridioforme
46                      MSK1_10_Collinsella_aerofaciens
47                           MSK17_80_Coprococcus_comes
48                           MSK17_85_Coprococcus_comes
49                              TM466_Coprococcus_comes
50                        MSK18_68_Coprococcus_eutactus
51                           TM431_Coprococcus_eutactus
52                       MSK23_29_Dorea_formicigenerans
53                          TM359_Dorea_formicigenerans
54                           MSK16_48_Dorea_longicatena
55                           MSK16_91_Dorea_longicatena
56                           MSK17_11_Dorea_longicatena
57                           MSK20_75_Dorea_longicatena
58                          MSK15_110_Eggerthella_lenta
59              MSK22_16_Erysipelatoclostridium_ramosum
60                 TM494_Erysipelatoclostridium_ramosum
61                 TM497_Erysipelatoclostridium_ramosum
62  EP-10S-DFI-2-03_Erysipelatoclostridium_unclassified
63         MSK23_31_Erysipelatoclostridium_unclassified
64               210119-DFI.3.32_Escherichia_fergusonii
65                         TM413_Escherichia_fergusonii
66                          TM75_Escherichia_fergusonii
67                         MSK22_19_Eubacterium_rectale
68                            TM426_Eubacterium_rectale
69         210226-DFI.5.42_Faecalibacterium_prausnitzii
70              EP-SM-12S-S02_Faecalicatena_fissicatena
71                      TM537_Faecalicatena_fissicatena
72             MSK19_10_Fusicatenibacter_saccharivorans
73                       MSK21_54_Holdemanella_biformis
74                  MSK17_41_Intestinibacter_bartlettii
75            210329-SL.2.13_Lachnoclostridium_pacaense
76                           TM1_Megamonas_unclassified
77                            TM67_Megasphaera_elsdenii
78             210120-DFI.3.53_Odoribacter_splanchnicus
79           210120-DFI.1.37_Parabacteroides_distasonis
80           210120-DFI.2.49_Parabacteroides_distasonis
81                  MSK22_22_Parabacteroides_distasonis
82             210309-DFI.4.154A_Parabacteroides_merdae
83                     MSK2_62_Pediococcus_acidilactici
84                     MSK2_11_Pediococcus_unclassified
85                            MSK21_28_Prevotella_copri
86                            MSK21_56_Prevotella_copri
87                            MSK21_64_Prevotella_copri
88                            MSK21_66_Prevotella_copri
89                            TM343_Ruminococcus_faecis
90                         MSK22_24_Ruminococcus_gnavus
91                         MSK23_60_Ruminococcus_gnavus
92                       MSK17_20_Ruminococcus_lactaris
93                          MSK15_1_Salmonella_enterica
94                      MSK1_19_Sellimonas_intestinalis
95                             MSK19_96_Shigella_sonnei
96                              MSK5_13_Shigella_sonnei
97                     TM356_Staphylococcus_epidermidis
98                        MSK15_92_Streptococcus_oralis
99                                MSK19_28_unclassified
100                     FC22_134_Weissella_unclassified
                                species
1          Acidaminococcus unclassified
2                  Alistipes finegoldii
3                   Anaerostipes hadrus
4                   Anaerostipes hadrus
5                   Anaerostipes hadrus
6                   Anaerostipes hadrus
7                   Anaerostipes hadrus
8                   Anaerostipes hadrus
9                 Bacillus unclassified
10                    Bacteroides dorei
11                 Bacteroides fragilis
12             Bacteroides massiliensis
13               Bacteroides salyersiae
14                Bacteroides stercoris
15         Bacteroides thetaiotaomicron
16         Bacteroides thetaiotaomicron
17         Bacteroides thetaiotaomicron
18         Bacteroides thetaiotaomicron
19             Bacteroides unclassified
20             Bacteroides unclassified
21                Bacteroides uniformis
22                 Bacteroides vulgatus
23         Bifidobacterium adolescentis
24         Bifidobacterium adolescentis
25         Bifidobacterium adolescentis
26         Bifidobacterium adolescentis
27               Bifidobacterium longum
28         Bifidobacterium unclassified
29                       Blautia faecis
30                     Blautia hansenii
31                         Blautia luti
32                         Blautia luti
33                        Blautia obeum
34                        Blautia obeum
35                     Blautia producta
36                     Blautia schinkii
37                     Blautia wexlerae
38                     Blautia wexlerae
39                     Blautia wexlerae
40                     Blautia wexlerae
41                     Blautia wexlerae
42                     Blautia wexlerae
43            Catenibacterium mitsuokai
44                Clostridium aldenense
45          Clostridium clostridioforme
46              Collinsella aerofaciens
47                    Coprococcus comes
48                    Coprococcus comes
49                    Coprococcus comes
50                 Coprococcus eutactus
51                 Coprococcus eutactus
52                Dorea formicigenerans
53                Dorea formicigenerans
54                    Dorea longicatena
55                    Dorea longicatena
56                    Dorea longicatena
57                    Dorea longicatena
58                    Eggerthella lenta
59       Erysipelatoclostridium ramosum
60       Erysipelatoclostridium ramosum
61       Erysipelatoclostridium ramosum
62  Erysipelatoclostridium unclassified
63  Erysipelatoclostridium unclassified
64               Escherichia fergusonii
65               Escherichia fergusonii
66               Escherichia fergusonii
67                  Eubacterium rectale
68                  Eubacterium rectale
69         Faecalibacterium prausnitzii
70            Faecalicatena fissicatena
71            Faecalicatena fissicatena
72      Fusicatenibacter saccharivorans
73                Holdemanella biformis
74           Intestinibacter bartlettii
75           Lachnoclostridium pacaense
76               Megamonas unclassified
77                 Megasphaera elsdenii
78             Odoribacter splanchnicus
79           Parabacteroides distasonis
80           Parabacteroides distasonis
81           Parabacteroides distasonis
82               Parabacteroides merdae
83             Pediococcus acidilactici
84             Pediococcus unclassified
85                     Prevotella copri
86                     Prevotella copri
87                     Prevotella copri
88                     Prevotella copri
89                  Ruminococcus faecis
90                  Ruminococcus gnavus
91                  Ruminococcus gnavus
92                Ruminococcus lactaris
93                  Salmonella enterica
94              Sellimonas intestinalis
95                      Shigella sonnei
96                      Shigella sonnei
97           Staphylococcus epidermidis
98                 Streptococcus oralis
99                         unclassified
100              Weissella unclassified</code></pre>
<pre class="r"><code># looking into the clusters
kmeans_cls[kmeans_cls==90]</code></pre>
<pre><code>210309-MSK.14.97_Blautia_producta     210309-SL2.02_Blautia_marasmi 
                               90                                90 
            MSK16_34_Blautia_luti                TM475_Blautia_luti 
                               90                                90 
            MSK22_60_Blautia_luti            TM122_Blautia_schinkii 
                               90                                90 
           TM123_Blautia_schinkii            TM124_Blautia_schinkii 
                               90                                90 
           TM136_Blautia_schinkii            TM126_Blautia_schinkii 
                               90                                90 
             MSK20_9_Blautia_luti             MSK20_85_Blautia_luti 
                               90                                90 
           TM141_Blautia_hansenii            TM118_Blautia_producta 
                               90                                90 
           TM119_Blautia_producta          TM164_Blautia_caecimuris 
                               90                                90 </code></pre>
<pre class="r"><code># searching
df_cls &lt;- data.frame(kmeans_cls)
df_cls &lt;- tibble::rownames_to_column(df_cls, var=&quot;id&quot;)
colnames(df_cls)</code></pre>
<pre><code>[1] &quot;id&quot;         &quot;kmeans_cls&quot;</code></pre>
<pre class="r"><code>df_cls %&gt;% filter(str_detect(id, &quot;Bacteroides_fragilis&quot;))</code></pre>
<pre><code>                             id kmeans_cls
1 MSK16_53_Bacteroides_fragilis          3
2 MSK16_58_Bacteroides_fragilis          3
3    TM242_Bacteroides_fragilis          3
4    TM389_Bacteroides_fragilis          3
5    TM350_Bacteroides_fragilis         21
6    TM430_Bacteroides_fragilis         21</code></pre>
</div>
<div id="comparison-of-100-strain-sets" class="section level2">
<h2>4. Comparison of 100 strain sets</h2>
<p>Now we will calculate NRI (Net Relatedness Index) and NTI (Nearest Taxon Index) for our different communities. Rembember, negative NRI and NTI values indicate a high level of phylogenetic overdispersion, and positive NRI and NTI values indicate phylogenetic clustering.<br />
- Source: <a href="http://ib.berkeley.edu/courses/ib200/2016/labs/14/lab14.R" class="uri">http://ib.berkeley.edu/courses/ib200/2016/labs/14/lab14.R</a><br />
- Source2: <a href="http://ib.berkeley.edu/courses/ib200b/labs/lab16/ib200b_lab16_phylocomm.pdf" class="uri">http://ib.berkeley.edu/courses/ib200b/labs/lab16/ib200b_lab16_phylocomm.pdf</a><br />
- Paper: Horner-Devine, M. C., &amp; Bohannan, B. J. (2006). Phylogenetic clustering and overdispersion in bacterial communities. Ecology, 87(sp7), S100-S108. <a href="https://esajournals.onlinelibrary.wiley.com/doi/10.1890/0012-9658%282006%2987%5B100%3APCAOIB%5D2.0.CO%3B2" class="uri">https://esajournals.onlinelibrary.wiley.com/doi/10.1890/0012-9658%282006%2987%5B100%3APCAOIB%5D2.0.CO%3B2</a><br />
</p>
<pre class="r"><code># PICANTE R package
# install.packages(&quot;picante&quot;)
library(picante)

# strain list for 2 different methods
# (Method 1) UMAP method
strains_umap &lt;- center_strains
# (Method 2) Phylogenetic tree method
strains_phy &lt;- phy_id_list

# Let&#39;s visualize our data. Now let&#39;s see how taxa from the 2 communities from different methods arranged on the tree. 
phy_tree</code></pre>
<pre><code>
Phylogenetic tree with 850 tips and 849 internal nodes.

Tip labels:
  210119-DFI.3.32_Escherichia_fergusonii, EP-10S-DFI-2-08_Escherichia_fergusonii, MSK5_13_Shigella_sonnei, MSK17_69_Escherichia_fergusonii, 210309-SL2.01_Escherichia_fergusonii, MSK14_41_Escherichia_fergusonii, ...

Rooted; includes branch lengths.</code></pre>
<pre class="r"><code>phy_tree$edge</code></pre>
<pre><code>        [,1] [,2]
   [1,]  851  852
   [2,]  852    1
   [3,]  852  853
   [4,]  853  854
   [5,]  854  855
   [6,]  855    2
   [7,]  855  856
   [8,]  856  857
   [9,]  857  858
  [10,]  858    3
  [11,]  858    4
  [12,]  857    5
  [13,]  856  859
  [14,]  859  860
  [15,]  860  861
  [16,]  861  862
  [17,]  862  863
  [18,]  863  864
  [19,]  864    6
  [20,]  864    7
  [21,]  863    8
  [22,]  862    9
  [23,]  861   10
  [24,]  860  865
  [25,]  865  866
  [26,]  866  867
  [27,]  867  868
  [28,]  868  869
  [29,]  869  870
  [30,]  870  871
  [31,]  871   11
  [32,]  871  872
  [33,]  872   12
  [34,]  872   13
  [35,]  870   14
  [36,]  869   15
  [37,]  868  873
  [38,]  873  874
  [39,]  874   16
  [40,]  874   17
  [41,]  873   18
  [42,]  867   19
  [43,]  866   20
  [44,]  865  875
  [45,]  875   21
  [46,]  875   22
  [47,]  859  876
  [48,]  876  877
  [49,]  877  878
  [50,]  878   23
  [51,]  878   24
  [52,]  877   25
  [53,]  876  879
  [54,]  879  880
  [55,]  880  881
  [56,]  881  882
  [57,]  882  883
  [58,]  883  884
  [59,]  884  885
  [60,]  885  886
  [61,]  886  887
  [62,]  887  888
  [63,]  888  889
  [64,]  889  890
  [65,]  890  891
  [66,]  891  892
  [67,]  892  893
  [68,]  893  894
  [69,]  894  895
  [70,]  895  896
  [71,]  896  897
  [72,]  897  898
  [73,]  898   26
  [74,]  898  899
  [75,]  899  900
  [76,]  900  901
  [77,]  901  902
  [78,]  902  903
  [79,]  903  904
  [80,]  904  905
  [81,]  905   27
  [82,]  905   28
  [83,]  904   29
  [84,]  903   30
  [85,]  902   31
  [86,]  901   32
  [87,]  900  906
  [88,]  906   33
  [89,]  906   34
  [90,]  899  907
  [91,]  907   35
  [92,]  907  908
  [93,]  908   36
  [94,]  908  909
  [95,]  909   37
  [96,]  909   38
  [97,]  897   39
  [98,]  896   40
  [99,]  895  910
 [100,]  910   41
 [101,]  910   42
 [102,]  894   43
 [103,]  893   44
 [104,]  892  911
 [105,]  911  912
 [106,]  912  913
 [107,]  913  914
 [108,]  914   45
 [109,]  914   46
 [110,]  913   47
 [111,]  912   48
 [112,]  911   49
 [113,]  891   50
 [114,]  890   51
 [115,]  889   52
 [116,]  888   53
 [117,]  887   54
 [118,]  886  915
 [119,]  915  916
 [120,]  916  917
 [121,]  917  918
 [122,]  918  919
 [123,]  919  920
 [124,]  920  921
 [125,]  921   55
 [126,]  921   56
 [127,]  920  922
 [128,]  922  923
 [129,]  923   57
 [130,]  923   58
 [131,]  922   59
 [132,]  919  924
 [133,]  924  925
 [134,]  925  926
 [135,]  926  927
 [136,]  927  928
 [137,]  928  929
 [138,]  929  930
 [139,]  930  931
 [140,]  931  932
 [141,]  932  933
 [142,]  933  934
 [143,]  934  935
 [144,]  935  936
 [145,]  936  937
 [146,]  937   60
 [147,]  937  938
 [148,]  938   61
 [149,]  938  939
 [150,]  939   62
 [151,]  939  940
 [152,]  940  941
 [153,]  941  942
 [154,]  942   63
 [155,]  942   64
 [156,]  941   65
 [157,]  940  943
 [158,]  943   66
 [159,]  943   67
 [160,]  936   68
 [161,]  935   69
 [162,]  934  944
 [163,]  944  945
 [164,]  945  946
 [165,]  946  947
 [166,]  947  948
 [167,]  948  949
 [168,]  949  950
 [169,]  950   70
 [170,]  950  951
 [171,]  951  952
 [172,]  952   71
 [173,]  952   72
 [174,]  951   73
 [175,]  949  953
 [176,]  953  954
 [177,]  954  955
 [178,]  955  956
 [179,]  956  957
 [180,]  957  958
 [181,]  958  959
 [182,]  959  960
 [183,]  960   74
 [184,]  960  961
 [185,]  961   75
 [186,]  961   76
 [187,]  959   77
 [188,]  958   78
 [189,]  957   79
 [190,]  956   80
 [191,]  955  962
 [192,]  962  963
 [193,]  963  964
 [194,]  964  965
 [195,]  965  966
 [196,]  966  967
 [197,]  967  968
 [198,]  968   81
 [199,]  968   82
 [200,]  967   83
 [201,]  966   84
 [202,]  965   85
 [203,]  964   86
 [204,]  963   87
 [205,]  962   88
 [206,]  954   89
 [207,]  953   90
 [208,]  948   91
 [209,]  947   92
 [210,]  946  969
 [211,]  969   93
 [212,]  969   94
 [213,]  945  970
 [214,]  970  971
 [215,]  971   95
 [216,]  971   96
 [217,]  970   97
 [218,]  944   98
 [219,]  933   99
 [220,]  932  972
 [221,]  972  973
 [222,]  973  100
 [223,]  973  974
 [224,]  974  975
 [225,]  975  101
 [226,]  975  102
 [227,]  974  976
 [228,]  976  103
 [229,]  976  977
 [230,]  977  978
 [231,]  978  104
 [232,]  978  105
 [233,]  977  106
 [234,]  972  107
 [235,]  931  979
 [236,]  979  980
 [237,]  980  981
 [238,]  981  108
 [239,]  981  982
 [240,]  982  109
 [241,]  982  110
 [242,]  980  983
 [243,]  983  984
 [244,]  984  111
 [245,]  984  112
 [246,]  983  113
 [247,]  979  114
 [248,]  930  985
 [249,]  985  986
 [250,]  986  987
 [251,]  987  988
 [252,]  988  989
 [253,]  989  115
 [254,]  989  116
 [255,]  988  117
 [256,]  987  990
 [257,]  990  991
 [258,]  991  118
 [259,]  991  992
 [260,]  992  119
 [261,]  992  120
 [262,]  990  121
 [263,]  986  122
 [264,]  985  123
 [265,]  929  124
 [266,]  928  993
 [267,]  993  994
 [268,]  994  995
 [269,]  995  125
 [270,]  995  126
 [271,]  994  127
 [272,]  993  128
 [273,]  927  129
 [274,]  926  996
 [275,]  996  997
 [276,]  997  130
 [277,]  997  131
 [278,]  996  132
 [279,]  925  998
 [280,]  998  999
 [281,]  999 1000
 [282,] 1000 1001
 [283,] 1001  133
 [284,] 1001  134
 [285,] 1000 1002
 [286,] 1002  135
 [287,] 1002 1003
 [288,] 1003  136
 [289,] 1003  137
 [290,]  999 1004
 [291,] 1004  138
 [292,] 1004  139
 [293,]  998  140
 [294,]  924  141
 [295,]  918  142
 [296,]  917 1005
 [297,] 1005 1006
 [298,] 1006 1007
 [299,] 1007 1008
 [300,] 1008  143
 [301,] 1008 1009
 [302,] 1009 1010
 [303,] 1010  144
 [304,] 1010  145
 [305,] 1009  146
 [306,] 1007 1011
 [307,] 1011 1012
 [308,] 1012  147
 [309,] 1012 1013
 [310,] 1013  148
 [311,] 1013 1014
 [312,] 1014  149
 [313,] 1014  150
 [314,] 1011 1015
 [315,] 1015 1016
 [316,] 1016 1017
 [317,] 1017  151
 [318,] 1017 1018
 [319,] 1018  152
 [320,] 1018 1019
 [321,] 1019 1020
 [322,] 1020  153
 [323,] 1020  154
 [324,] 1019 1021
 [325,] 1021 1022
 [326,] 1022  155
 [327,] 1022  156
 [328,] 1021  157
 [329,] 1016  158
 [330,] 1015  159
 [331,] 1006 1023
 [332,] 1023 1024
 [333,] 1024 1025
 [334,] 1025 1026
 [335,] 1026 1027
 [336,] 1027 1028
 [337,] 1028 1029
 [338,] 1029  160
 [339,] 1029 1030
 [340,] 1030 1031
 [341,] 1031  161
 [342,] 1031  162
 [343,] 1030  163
 [344,] 1028  164
 [345,] 1027 1032
 [346,] 1032 1033
 [347,] 1033 1034
 [348,] 1034 1035
 [349,] 1035 1036
 [350,] 1036 1037
 [351,] 1037 1038
 [352,] 1038 1039
 [353,] 1039  165
 [354,] 1039  166
 [355,] 1038  167
 [356,] 1037  168
 [357,] 1036  169
 [358,] 1035  170
 [359,] 1034  171
 [360,] 1033  172
 [361,] 1032  173
 [362,] 1026  174
 [363,] 1025  175
 [364,] 1024  176
 [365,] 1023  177
 [366,] 1005  178
 [367,]  916 1040
 [368,] 1040 1041
 [369,] 1041 1042
 [370,] 1042 1043
 [371,] 1043 1044
 [372,] 1044 1045
 [373,] 1045  179
 [374,] 1045  180
 [375,] 1044  181
 [376,] 1043  182
 [377,] 1042  183
 [378,] 1041 1046
 [379,] 1046  184
 [380,] 1046  185
 [381,] 1040 1047
 [382,] 1047 1048
 [383,] 1048 1049
 [384,] 1049 1050
 [385,] 1050 1051
 [386,] 1051 1052
 [387,] 1052 1053
 [388,] 1053  186
 [389,] 1053  187
 [390,] 1052 1054
 [391,] 1054 1055
 [392,] 1055 1056
 [393,] 1056  188
 [394,] 1056  189
 [395,] 1055  190
 [396,] 1054  191
 [397,] 1051  192
 [398,] 1050 1057
 [399,] 1057 1058
 [400,] 1058 1059
 [401,] 1059 1060
 [402,] 1060 1061
 [403,] 1061 1062
 [404,] 1062 1063
 [405,] 1063 1064
 [406,] 1064  193
 [407,] 1064  194
 [408,] 1063  195
 [409,] 1062  196
 [410,] 1061  197
 [411,] 1060  198
 [412,] 1059  199
 [413,] 1058  200
 [414,] 1057  201
 [415,] 1049 1065
 [416,] 1065 1066
 [417,] 1066 1067
 [418,] 1067 1068
 [419,] 1068 1069
 [420,] 1069  202
 [421,] 1069  203
 [422,] 1068  204
 [423,] 1067  205
 [424,] 1066 1070
 [425,] 1070  206
 [426,] 1070  207
 [427,] 1065  208
 [428,] 1048  209
 [429,] 1047  210
 [430,]  915 1071
 [431,] 1071 1072
 [432,] 1072 1073
 [433,] 1073 1074
 [434,] 1074  211
 [435,] 1074 1075
 [436,] 1075 1076
 [437,] 1076 1077
 [438,] 1077 1078
 [439,] 1078  212
 [440,] 1078  213
 [441,] 1077  214
 [442,] 1076  215
 [443,] 1075  216
 [444,] 1073 1079
 [445,] 1079 1080
 [446,] 1080 1081
 [447,] 1081 1082
 [448,] 1082 1083
 [449,] 1083 1084
 [450,] 1084 1085
 [451,] 1085 1086
 [452,] 1086 1087
 [453,] 1087  217
 [454,] 1087  218
 [455,] 1086 1088
 [456,] 1088  219
 [457,] 1088  220
 [458,] 1085 1089
 [459,] 1089 1090
 [460,] 1090 1091
 [461,] 1091  221
 [462,] 1091  222
 [463,] 1090 1092
 [464,] 1092 1093
 [465,] 1093 1094
 [466,] 1094  223
 [467,] 1094  224
 [468,] 1093  225
 [469,] 1092 1095
 [470,] 1095  226
 [471,] 1095  227
 [472,] 1089  228
 [473,] 1084  229
 [474,] 1083  230
 [475,] 1082  231
 [476,] 1081 1096
 [477,] 1096  232
 [478,] 1096 1097
 [479,] 1097  233
 [480,] 1097 1098
 [481,] 1098  234
 [482,] 1098 1099
 [483,] 1099 1100
 [484,] 1100  235
 [485,] 1100  236
 [486,] 1099  237
 [487,] 1080  238
 [488,] 1079 1101
 [489,] 1101 1102
 [490,] 1102 1103
 [491,] 1103 1104
 [492,] 1104 1105
 [493,] 1105 1106
 [494,] 1106 1107
 [495,] 1107 1108
 [496,] 1108 1109
 [497,] 1109 1110
 [498,] 1110 1111
 [499,] 1111 1112
 [500,] 1112 1113
 [501,] 1113  239
 [502,] 1113 1114
 [503,] 1114  240
 [504,] 1114  241
 [505,] 1112 1115
 [506,] 1115 1116
 [507,] 1116 1117
 [508,] 1117  242
 [509,] 1117  243
 [510,] 1116  244
 [511,] 1115  245
 [512,] 1111  246
 [513,] 1110  247
 [514,] 1109  248
 [515,] 1108  249
 [516,] 1107  250
 [517,] 1106 1118
 [518,] 1118 1119
 [519,] 1119  251
 [520,] 1119  252
 [521,] 1118  253
 [522,] 1105 1120
 [523,] 1120  254
 [524,] 1120  255
 [525,] 1104 1121
 [526,] 1121 1122
 [527,] 1122  256
 [528,] 1122 1123
 [529,] 1123 1124
 [530,] 1124  257
 [531,] 1124  258
 [532,] 1123  259
 [533,] 1121  260
 [534,] 1103 1125
 [535,] 1125  261
 [536,] 1125  262
 [537,] 1102 1126
 [538,] 1126 1127
 [539,] 1127 1128
 [540,] 1128  263
 [541,] 1128  264
 [542,] 1127  265
 [543,] 1126  266
 [544,] 1101 1129
 [545,] 1129 1130
 [546,] 1130 1131
 [547,] 1131 1132
 [548,] 1132 1133
 [549,] 1133  267
 [550,] 1133  268
 [551,] 1132  269
 [552,] 1131 1134
 [553,] 1134 1135
 [554,] 1135 1136
 [555,] 1136  270
 [556,] 1136  271
 [557,] 1135  272
 [558,] 1134  273
 [559,] 1130  274
 [560,] 1129 1137
 [561,] 1137 1138
 [562,] 1138 1139
 [563,] 1139  275
 [564,] 1139  276
 [565,] 1138  277
 [566,] 1137  278
 [567,] 1072 1140
 [568,] 1140  279
 [569,] 1140  280
 [570,] 1071 1141
 [571,] 1141  281
 [572,] 1141 1142
 [573,] 1142 1143
 [574,] 1143 1144
 [575,] 1144 1145
 [576,] 1145 1146
 [577,] 1146 1147
 [578,] 1147 1148
 [579,] 1148 1149
 [580,] 1149 1150
 [581,] 1150 1151
 [582,] 1151 1152
 [583,] 1152 1153
 [584,] 1153 1154
 [585,] 1154 1155
 [586,] 1155 1156
 [587,] 1156 1157
 [588,] 1157 1158
 [589,] 1158 1159
 [590,] 1159 1160
 [591,] 1160 1161
 [592,] 1161 1162
 [593,] 1162 1163
 [594,] 1163  282
 [595,] 1163  283
 [596,] 1162  284
 [597,] 1161  285
 [598,] 1160  286
 [599,] 1159  287
 [600,] 1158  288
 [601,] 1157  289
 [602,] 1156  290
 [603,] 1155  291
 [604,] 1154  292
 [605,] 1153  293
 [606,] 1152  294
 [607,] 1151  295
 [608,] 1150  296
 [609,] 1149  297
 [610,] 1148  298
 [611,] 1147  299
 [612,] 1146  300
 [613,] 1145 1164
 [614,] 1164  301
 [615,] 1164  302
 [616,] 1144 1165
 [617,] 1165  303
 [618,] 1165  304
 [619,] 1143 1166
 [620,] 1166 1167
 [621,] 1167 1168
 [622,] 1168 1169
 [623,] 1169 1170
 [624,] 1170 1171
 [625,] 1171 1172
 [626,] 1172 1173
 [627,] 1173  305
 [628,] 1173  306
 [629,] 1172  307
 [630,] 1171  308
 [631,] 1170  309
 [632,] 1169  310
 [633,] 1168 1174
 [634,] 1174  311
 [635,] 1174  312
 [636,] 1167  313
 [637,] 1166  314
 [638,] 1142  315
 [639,]  885  316
 [640,]  884 1175
 [641,] 1175 1176
 [642,] 1176 1177
 [643,] 1177  317
 [644,] 1177  318
 [645,] 1176  319
 [646,] 1175 1178
 [647,] 1178  320
 [648,] 1178  321
 [649,]  883 1179
 [650,] 1179  322
 [651,] 1179 1180
 [652,] 1180  323
 [653,] 1180 1181
 [654,] 1181  324
 [655,] 1181 1182
 [656,] 1182  325
 [657,] 1182  326
 [658,]  882 1183
 [659,] 1183 1184
 [660,] 1184 1185
 [661,] 1185 1186
 [662,] 1186  327
 [663,] 1186  328
 [664,] 1185  329
 [665,] 1184  330
 [666,] 1183 1187
 [667,] 1187 1188
 [668,] 1188 1189
 [669,] 1189 1190
 [670,] 1190 1191
 [671,] 1191 1192
 [672,] 1192 1193
 [673,] 1193 1194
 [674,] 1194  331
 [675,] 1194  332
 [676,] 1193 1195
 [677,] 1195  333
 [678,] 1195  334
 [679,] 1192  335
 [680,] 1191  336
 [681,] 1190 1196
 [682,] 1196 1197
 [683,] 1197 1198
 [684,] 1198 1199
 [685,] 1199  337
 [686,] 1199  338
 [687,] 1198 1200
 [688,] 1200  339
 [689,] 1200  340
 [690,] 1197 1201
 [691,] 1201  341
 [692,] 1201  342
 [693,] 1196 1202
 [694,] 1202  343
 [695,] 1202  344
 [696,] 1189 1203
 [697,] 1203 1204
 [698,] 1204 1205
 [699,] 1205  345
 [700,] 1205 1206
 [701,] 1206 1207
 [702,] 1207 1208
 [703,] 1208 1209
 [704,] 1209 1210
 [705,] 1210 1211
 [706,] 1211 1212
 [707,] 1212 1213
 [708,] 1213 1214
 [709,] 1214 1215
 [710,] 1215  346
 [711,] 1215  347
 [712,] 1214  348
 [713,] 1213  349
 [714,] 1212  350
 [715,] 1211  351
 [716,] 1210  352
 [717,] 1209  353
 [718,] 1208  354
 [719,] 1207 1216
 [720,] 1216  355
 [721,] 1216  356
 [722,] 1206  357
 [723,] 1204  358
 [724,] 1203 1217
 [725,] 1217 1218
 [726,] 1218 1219
 [727,] 1219 1220
 [728,] 1220 1221
 [729,] 1221 1222
 [730,] 1222 1223
 [731,] 1223 1224
 [732,] 1224 1225
 [733,] 1225 1226
 [734,] 1226 1227
 [735,] 1227 1228
 [736,] 1228 1229
 [737,] 1229 1230
 [738,] 1230 1231
 [739,] 1231 1232
 [740,] 1232 1233
 [741,] 1233 1234
 [742,] 1234 1235
 [743,] 1235 1236
 [744,] 1236 1237
 [745,] 1237 1238
 [746,] 1238 1239
 [747,] 1239 1240
 [748,] 1240 1241
 [749,] 1241 1242
 [750,] 1242 1243
 [751,] 1243 1244
 [752,] 1244  359
 [753,] 1244  360
 [754,] 1243  361
 [755,] 1242  362
 [756,] 1241  363
 [757,] 1240  364
 [758,] 1239  365
 [759,] 1238  366
 [760,] 1237  367
 [761,] 1236  368
 [762,] 1235  369
 [763,] 1234  370
 [764,] 1233  371
 [765,] 1232  372
 [766,] 1231  373
 [767,] 1230  374
 [768,] 1229  375
 [769,] 1228  376
 [770,] 1227  377
 [771,] 1226  378
 [772,] 1225  379
 [773,] 1224  380
 [774,] 1223  381
 [775,] 1222 1245
 [776,] 1245  382
 [777,] 1245 1246
 [778,] 1246  383
 [779,] 1246 1247
 [780,] 1247  384
 [781,] 1247 1248
 [782,] 1248  385
 [783,] 1248  386
 [784,] 1221  387
 [785,] 1220  388
 [786,] 1219 1249
 [787,] 1249  389
 [788,] 1249  390
 [789,] 1218  391
 [790,] 1217  392
 [791,] 1188 1250
 [792,] 1250 1251
 [793,] 1251  393
 [794,] 1251 1252
 [795,] 1252  394
 [796,] 1252  395
 [797,] 1250 1253
 [798,] 1253 1254
 [799,] 1254 1255
 [800,] 1255 1256
 [801,] 1256  396
 [802,] 1256  397
 [803,] 1255 1257
 [804,] 1257 1258
 [805,] 1258 1259
 [806,] 1259 1260
 [807,] 1260 1261
 [808,] 1261 1262
 [809,] 1262 1263
 [810,] 1263 1264
 [811,] 1264 1265
 [812,] 1265 1266
 [813,] 1266  398
 [814,] 1266  399
 [815,] 1265  400
 [816,] 1264  401
 [817,] 1263  402
 [818,] 1262  403
 [819,] 1261  404
 [820,] 1260  405
 [821,] 1259  406
 [822,] 1258  407
 [823,] 1257 1267
 [824,] 1267  408
 [825,] 1267  409
 [826,] 1254 1268
 [827,] 1268  410
 [828,] 1268 1269
 [829,] 1269 1270
 [830,] 1270 1271
 [831,] 1271 1272
 [832,] 1272 1273
 [833,] 1273 1274
 [834,] 1274 1275
 [835,] 1275 1276
 [836,] 1276 1277
 [837,] 1277 1278
 [838,] 1278 1279
 [839,] 1279 1280
 [840,] 1280 1281
 [841,] 1281  411
 [842,] 1281  412
 [843,] 1280  413
 [844,] 1279  414
 [845,] 1278  415
 [846,] 1277 1282
 [847,] 1282 1283
 [848,] 1283 1284
 [849,] 1284 1285
 [850,] 1285 1286
 [851,] 1286  416
 [852,] 1286  417
 [853,] 1285  418
 [854,] 1284  419
 [855,] 1283  420
 [856,] 1282  421
 [857,] 1276  422
 [858,] 1275  423
 [859,] 1274  424
 [860,] 1273  425
 [861,] 1272  426
 [862,] 1271  427
 [863,] 1270 1287
 [864,] 1287 1288
 [865,] 1288 1289
 [866,] 1289 1290
 [867,] 1290 1291
 [868,] 1291 1292
 [869,] 1292  428
 [870,] 1292  429
 [871,] 1291  430
 [872,] 1290  431
 [873,] 1289  432
 [874,] 1288  433
 [875,] 1287  434
 [876,] 1269  435
 [877,] 1253 1293
 [878,] 1293 1294
 [879,] 1294  436
 [880,] 1294  437
 [881,] 1293 1295
 [882,] 1295 1296
 [883,] 1296  438
 [884,] 1296  439
 [885,] 1295  440
 [886,] 1187 1297
 [887,] 1297 1298
 [888,] 1298 1299
 [889,] 1299 1300
 [890,] 1300 1301
 [891,] 1301  441
 [892,] 1301  442
 [893,] 1300  443
 [894,] 1299 1302
 [895,] 1302  444
 [896,] 1302  445
 [897,] 1298 1303
 [898,] 1303  446
 [899,] 1303  447
 [900,] 1297 1304
 [901,] 1304 1305
 [902,] 1305 1306
 [903,] 1306 1307
 [904,] 1307  448
 [905,] 1307 1308
 [906,] 1308 1309
 [907,] 1309 1310
 [908,] 1310  449
 [909,] 1310  450
 [910,] 1309  451
 [911,] 1308  452
 [912,] 1306 1311
 [913,] 1311 1312
 [914,] 1312 1313
 [915,] 1313 1314
 [916,] 1314 1315
 [917,] 1315 1316
 [918,] 1316  453
 [919,] 1316  454
 [920,] 1315  455
 [921,] 1314  456
 [922,] 1313  457
 [923,] 1312  458
 [924,] 1311  459
 [925,] 1305 1317
 [926,] 1317  460
 [927,] 1317  461
 [928,] 1304  462
 [929,]  881 1318
 [930,] 1318 1319
 [931,] 1319 1320
 [932,] 1320 1321
 [933,] 1321 1322
 [934,] 1322 1323
 [935,] 1323 1324
 [936,] 1324 1325
 [937,] 1325 1326
 [938,] 1326 1327
 [939,] 1327 1328
 [940,] 1328 1329
 [941,] 1329 1330
 [942,] 1330 1331
 [943,] 1331 1332
 [944,] 1332 1333
 [945,] 1333 1334
 [946,] 1334 1335
 [947,] 1335 1336
 [948,] 1336 1337
 [949,] 1337 1338
 [950,] 1338 1339
 [951,] 1339 1340
 [952,] 1340 1341
 [953,] 1341 1342
 [954,] 1342 1343
 [955,] 1343 1344
 [956,] 1344 1345
 [957,] 1345 1346
 [958,] 1346 1347
 [959,] 1347 1348
 [960,] 1348 1349
 [961,] 1349 1350
 [962,] 1350 1351
 [963,] 1351 1352
 [964,] 1352 1353
 [965,] 1353  463
 [966,] 1353  464
 [967,] 1352  465
 [968,] 1351  466
 [969,] 1350  467
 [970,] 1349  468
 [971,] 1348  469
 [972,] 1347  470
 [973,] 1346  471
 [974,] 1345  472
 [975,] 1344  473
 [976,] 1343  474
 [977,] 1342  475
 [978,] 1341  476
 [979,] 1340  477
 [980,] 1339  478
 [981,] 1338  479
 [982,] 1337  480
 [983,] 1336  481
 [984,] 1335  482
 [985,] 1334  483
 [986,] 1333  484
 [987,] 1332  485
 [988,] 1331  486
 [989,] 1330  487
 [990,] 1329  488
 [991,] 1328  489
 [992,] 1327  490
 [993,] 1326  491
 [994,] 1325  492
 [995,] 1324  493
 [996,] 1323 1354
 [997,] 1354 1355
 [998,] 1355 1356
 [999,] 1356 1357
[1000,] 1357 1358
[1001,] 1358  494
[1002,] 1358  495
[1003,] 1357  496
[1004,] 1356 1359
[1005,] 1359 1360
[1006,] 1360 1361
[1007,] 1361 1362
[1008,] 1362 1363
[1009,] 1363 1364
[1010,] 1364 1365
[1011,] 1365 1366
[1012,] 1366 1367
[1013,] 1367 1368
[1014,] 1368 1369
[1015,] 1369 1370
[1016,] 1370 1371
[1017,] 1371 1372
[1018,] 1372 1373
[1019,] 1373 1374
[1020,] 1374 1375
[1021,] 1375 1376
[1022,] 1376 1377
[1023,] 1377 1378
[1024,] 1378 1379
[1025,] 1379 1380
[1026,] 1380 1381
[1027,] 1381 1382
[1028,] 1382 1383
[1029,] 1383  497
[1030,] 1383  498
[1031,] 1382  499
[1032,] 1381  500
[1033,] 1380  501
[1034,] 1379  502
[1035,] 1378  503
[1036,] 1377  504
[1037,] 1376  505
[1038,] 1375  506
[1039,] 1374  507
[1040,] 1373  508
[1041,] 1372  509
[1042,] 1371  510
[1043,] 1370  511
[1044,] 1369  512
[1045,] 1368  513
[1046,] 1367  514
[1047,] 1366  515
[1048,] 1365  516
[1049,] 1364  517
[1050,] 1363  518
[1051,] 1362  519
[1052,] 1361  520
[1053,] 1360  521
[1054,] 1359  522
[1055,] 1355 1384
[1056,] 1384 1385
[1057,] 1385 1386
[1058,] 1386 1387
[1059,] 1387 1388
[1060,] 1388 1389
[1061,] 1389 1390
[1062,] 1390 1391
[1063,] 1391 1392
[1064,] 1392 1393
[1065,] 1393  523
[1066,] 1393  524
[1067,] 1392  525
[1068,] 1391  526
[1069,] 1390  527
[1070,] 1389  528
[1071,] 1388  529
[1072,] 1387  530
[1073,] 1386  531
[1074,] 1385  532
[1075,] 1384  533
[1076,] 1354  534
[1077,] 1322 1394
[1078,] 1394  535
[1079,] 1394  536
[1080,] 1321 1395
[1081,] 1395 1396
[1082,] 1396  537
[1083,] 1396  538
[1084,] 1395  539
[1085,] 1320 1397
[1086,] 1397 1398
[1087,] 1398 1399
[1088,] 1399 1400
[1089,] 1400 1401
[1090,] 1401 1402
[1091,] 1402 1403
[1092,] 1403  540
[1093,] 1403 1404
[1094,] 1404 1405
[1095,] 1405 1406
[1096,] 1406 1407
[1097,] 1407  541
[1098,] 1407  542
[1099,] 1406  543
[1100,] 1405  544
[1101,] 1404  545
[1102,] 1402  546
[1103,] 1401  547
[1104,] 1400  548
[1105,] 1399  549
[1106,] 1398  550
[1107,] 1397  551
[1108,] 1319 1408
[1109,] 1408 1409
[1110,] 1409 1410
[1111,] 1410 1411
[1112,] 1411 1412
[1113,] 1412 1413
[1114,] 1413  552
[1115,] 1413  553
[1116,] 1412  554
[1117,] 1411  555
[1118,] 1410  556
[1119,] 1409  557
[1120,] 1408  558
[1121,] 1318 1414
[1122,] 1414 1415
[1123,] 1415 1416
[1124,] 1416  559
[1125,] 1416 1417
[1126,] 1417  560
[1127,] 1417  561
[1128,] 1415  562
[1129,] 1414 1418
[1130,] 1418 1419
[1131,] 1419 1420
[1132,] 1420 1421
[1133,] 1421  563
[1134,] 1421  564
[1135,] 1420  565
[1136,] 1419  566
[1137,] 1418  567
[1138,]  880 1422
[1139,] 1422 1423
[1140,] 1423 1424
[1141,] 1424 1425
[1142,] 1425 1426
[1143,] 1426 1427
[1144,] 1427 1428
[1145,] 1428 1429
[1146,] 1429 1430
[1147,] 1430 1431
[1148,] 1431 1432
[1149,] 1432 1433
[1150,] 1433 1434
[1151,] 1434 1435
[1152,] 1435 1436
[1153,] 1436 1437
[1154,] 1437 1438
[1155,] 1438 1439
[1156,] 1439 1440
[1157,] 1440 1441
[1158,] 1441 1442
[1159,] 1442 1443
[1160,] 1443  568
[1161,] 1443  569
[1162,] 1442  570
[1163,] 1441  571
[1164,] 1440  572
[1165,] 1439  573
[1166,] 1438  574
[1167,] 1437  575
[1168,] 1436  576
[1169,] 1435 1444
[1170,] 1444 1445
[1171,] 1445 1446
[1172,] 1446 1447
[1173,] 1447  577
[1174,] 1447  578
[1175,] 1446 1448
[1176,] 1448  579
[1177,] 1448  580
[1178,] 1445  581
[1179,] 1444 1449
[1180,] 1449 1450
[1181,] 1450 1451
[1182,] 1451 1452
[1183,] 1452 1453
[1184,] 1453 1454
[1185,] 1454 1455
[1186,] 1455  582
[1187,] 1455  583
[1188,] 1454  584
[1189,] 1453  585
[1190,] 1452  586
[1191,] 1451  587
[1192,] 1450  588
[1193,] 1449  589
[1194,] 1434 1456
[1195,] 1456 1457
[1196,] 1457  590
[1197,] 1457  591
[1198,] 1456  592
[1199,] 1433  593
[1200,] 1432 1458
[1201,] 1458 1459
[1202,] 1459 1460
[1203,] 1460 1461
[1204,] 1461  594
[1205,] 1461  595
[1206,] 1460  596
[1207,] 1459  597
[1208,] 1458  598
[1209,] 1431  599
[1210,] 1430 1462
[1211,] 1462 1463
[1212,] 1463 1464
[1213,] 1464  600
[1214,] 1464 1465
[1215,] 1465 1466
[1216,] 1466 1467
[1217,] 1467 1468
[1218,] 1468 1469
[1219,] 1469 1470
[1220,] 1470 1471
[1221,] 1471 1472
[1222,] 1472 1473
[1223,] 1473 1474
[1224,] 1474  601
[1225,] 1474  602
[1226,] 1473  603
[1227,] 1472  604
[1228,] 1471  605
[1229,] 1470  606
[1230,] 1469  607
[1231,] 1468  608
[1232,] 1467  609
[1233,] 1466 1475
[1234,] 1475  610
[1235,] 1475 1476
[1236,] 1476 1477
[1237,] 1477 1478
[1238,] 1478 1479
[1239,] 1479  611
[1240,] 1479  612
[1241,] 1478 1480
[1242,] 1480  613
[1243,] 1480  614
[1244,] 1477  615
[1245,] 1476  616
[1246,] 1465 1481
[1247,] 1481  617
[1248,] 1481 1482
[1249,] 1482 1483
[1250,] 1483 1484
[1251,] 1484 1485
[1252,] 1485 1486
[1253,] 1486 1487
[1254,] 1487  618
[1255,] 1487  619
[1256,] 1486  620
[1257,] 1485  621
[1258,] 1484  622
[1259,] 1483  623
[1260,] 1482  624
[1261,] 1463 1488
[1262,] 1488 1489
[1263,] 1489  625
[1264,] 1489  626
[1265,] 1488  627
[1266,] 1462  628
[1267,] 1429 1490
[1268,] 1490  629
[1269,] 1490 1491
[1270,] 1491 1492
[1271,] 1492 1493
[1272,] 1493  630
[1273,] 1493  631
[1274,] 1492 1494
[1275,] 1494 1495
[1276,] 1495  632
[1277,] 1495  633
[1278,] 1494  634
[1279,] 1491 1496
[1280,] 1496  635
[1281,] 1496  636
[1282,] 1428 1497
[1283,] 1497 1498
[1284,] 1498 1499
[1285,] 1499 1500
[1286,] 1500 1501
[1287,] 1501 1502
[1288,] 1502  637
[1289,] 1502  638
[1290,] 1501  639
[1291,] 1500  640
[1292,] 1499  641
[1293,] 1498  642
[1294,] 1497 1503
[1295,] 1503  643
[1296,] 1503  644
[1297,] 1427 1504
[1298,] 1504 1505
[1299,] 1505 1506
[1300,] 1506 1507
[1301,] 1507  645
[1302,] 1507  646
[1303,] 1506  647
[1304,] 1505 1508
[1305,] 1508 1509
[1306,] 1509 1510
[1307,] 1510 1511
[1308,] 1511 1512
[1309,] 1512 1513
[1310,] 1513 1514
[1311,] 1514 1515
[1312,] 1515 1516
[1313,] 1516 1517
[1314,] 1517  648
[1315,] 1517  649
[1316,] 1516  650
[1317,] 1515  651
[1318,] 1514  652
[1319,] 1513  653
[1320,] 1512  654
[1321,] 1511 1518
[1322,] 1518 1519
[1323,] 1519 1520
[1324,] 1520  655
[1325,] 1520 1521
[1326,] 1521  656
[1327,] 1521  657
[1328,] 1519  658
[1329,] 1518  659
[1330,] 1510 1522
[1331,] 1522 1523
[1332,] 1523 1524
[1333,] 1524 1525
[1334,] 1525 1526
[1335,] 1526 1527
[1336,] 1527 1528
[1337,] 1528  660
[1338,] 1528  661
[1339,] 1527  662
[1340,] 1526  663
[1341,] 1525  664
[1342,] 1524  665
[1343,] 1523  666
[1344,] 1522  667
[1345,] 1509 1529
[1346,] 1529 1530
[1347,] 1530 1531
[1348,] 1531 1532
[1349,] 1532 1533
[1350,] 1533 1534
[1351,] 1534  668
[1352,] 1534  669
[1353,] 1533  670
[1354,] 1532  671
[1355,] 1531  672
[1356,] 1530  673
[1357,] 1529  674
[1358,] 1508  675
[1359,] 1504 1535
[1360,] 1535 1536
[1361,] 1536 1537
[1362,] 1537  676
[1363,] 1537  677
[1364,] 1536 1538
[1365,] 1538 1539
[1366,] 1539 1540
[1367,] 1540 1541
[1368,] 1541 1542
[1369,] 1542 1543
[1370,] 1543 1544
[1371,] 1544 1545
[1372,] 1545 1546
[1373,] 1546 1547
[1374,] 1547 1548
[1375,] 1548 1549
[1376,] 1549 1550
[1377,] 1550 1551
[1378,] 1551 1552
[1379,] 1552 1553
[1380,] 1553 1554
[1381,] 1554  678
[1382,] 1554  679
[1383,] 1553  680
[1384,] 1552  681
[1385,] 1551  682
[1386,] 1550  683
[1387,] 1549  684
[1388,] 1548  685
[1389,] 1547  686
[1390,] 1546  687
[1391,] 1545  688
[1392,] 1544  689
[1393,] 1543  690
[1394,] 1542  691
[1395,] 1541  692
[1396,] 1540  693
[1397,] 1539  694
[1398,] 1538  695
[1399,] 1535 1555
[1400,] 1555 1556
[1401,] 1556 1557
[1402,] 1557  696
[1403,] 1557  697
[1404,] 1556  698
[1405,] 1555 1558
[1406,] 1558  699
[1407,] 1558  700
[1408,] 1426 1559
[1409,] 1559 1560
[1410,] 1560 1561
[1411,] 1561 1562
[1412,] 1562 1563
[1413,] 1563 1564
[1414,] 1564  701
[1415,] 1564  702
[1416,] 1563  703
[1417,] 1562  704
[1418,] 1561 1565
[1419,] 1565 1566
[1420,] 1566 1567
[1421,] 1567  705
[1422,] 1567 1568
[1423,] 1568 1569
[1424,] 1569 1570
[1425,] 1570 1571
[1426,] 1571 1572
[1427,] 1572 1573
[1428,] 1573 1574
[1429,] 1574 1575
[1430,] 1575 1576
[1431,] 1576 1577
[1432,] 1577 1578
[1433,] 1578 1579
[1434,] 1579 1580
[1435,] 1580 1581
[1436,] 1581  706
[1437,] 1581  707
[1438,] 1580  708
[1439,] 1579  709
[1440,] 1578  710
[1441,] 1577 1582
[1442,] 1582 1583
[1443,] 1583 1584
[1444,] 1584 1585
[1445,] 1585 1586
[1446,] 1586 1587
[1447,] 1587 1588
[1448,] 1588 1589
[1449,] 1589 1590
[1450,] 1590 1591
[1451,] 1591 1592
[1452,] 1592 1593
[1453,] 1593 1594
[1454,] 1594 1595
[1455,] 1595  711
[1456,] 1595  712
[1457,] 1594  713
[1458,] 1593  714
[1459,] 1592  715
[1460,] 1591  716
[1461,] 1590  717
[1462,] 1589  718
[1463,] 1588  719
[1464,] 1587  720
[1465,] 1586  721
[1466,] 1585  722
[1467,] 1584  723
[1468,] 1583  724
[1469,] 1582  725
[1470,] 1576  726
[1471,] 1575 1596
[1472,] 1596 1597
[1473,] 1597 1598
[1474,] 1598 1599
[1475,] 1599 1600
[1476,] 1600 1601
[1477,] 1601 1602
[1478,] 1602 1603
[1479,] 1603 1604
[1480,] 1604 1605
[1481,] 1605 1606
[1482,] 1606 1607
[1483,] 1607 1608
[1484,] 1608 1609
[1485,] 1609 1610
[1486,] 1610 1611
[1487,] 1611 1612
[1488,] 1612 1613
[1489,] 1613 1614
[1490,] 1614 1615
[1491,] 1615 1616
[1492,] 1616 1617
[1493,] 1617 1618
[1494,] 1618 1619
[1495,] 1619 1620
[1496,] 1620 1621
[1497,] 1621 1622
[1498,] 1622 1623
[1499,] 1623 1624
[1500,] 1624 1625
[1501,] 1625  727
[1502,] 1625  728
[1503,] 1624  729
[1504,] 1623  730
[1505,] 1622  731
[1506,] 1621  732
[1507,] 1620  733
[1508,] 1619  734
[1509,] 1618  735
[1510,] 1617  736
[1511,] 1616  737
[1512,] 1615  738
[1513,] 1614  739
[1514,] 1613  740
[1515,] 1612  741
[1516,] 1611  742
[1517,] 1610  743
[1518,] 1609  744
[1519,] 1608  745
[1520,] 1607  746
[1521,] 1606  747
[1522,] 1605  748
[1523,] 1604  749
[1524,] 1603  750
[1525,] 1602  751
[1526,] 1601  752
[1527,] 1600  753
[1528,] 1599  754
[1529,] 1598  755
[1530,] 1597  756
[1531,] 1596 1626
[1532,] 1626  757
[1533,] 1626 1627
[1534,] 1627  758
[1535,] 1627  759
[1536,] 1574 1628
[1537,] 1628 1629
[1538,] 1629 1630
[1539,] 1630 1631
[1540,] 1631  760
[1541,] 1631  761
[1542,] 1630  762
[1543,] 1629  763
[1544,] 1628  764
[1545,] 1573  765
[1546,] 1572  766
[1547,] 1571 1632
[1548,] 1632 1633
[1549,] 1633 1634
[1550,] 1634 1635
[1551,] 1635 1636
[1552,] 1636 1637
[1553,] 1637 1638
[1554,] 1638  767
[1555,] 1638  768
[1556,] 1637  769
[1557,] 1636  770
[1558,] 1635  771
[1559,] 1634  772
[1560,] 1633  773
[1561,] 1632  774
[1562,] 1570 1639
[1563,] 1639 1640
[1564,] 1640 1641
[1565,] 1641 1642
[1566,] 1642 1643
[1567,] 1643 1644
[1568,] 1644 1645
[1569,] 1645 1646
[1570,] 1646 1647
[1571,] 1647  775
[1572,] 1647  776
[1573,] 1646  777
[1574,] 1645  778
[1575,] 1644  779
[1576,] 1643  780
[1577,] 1642  781
[1578,] 1641  782
[1579,] 1640  783
[1580,] 1639  784
[1581,] 1569  785
[1582,] 1568  786
[1583,] 1566  787
[1584,] 1565  788
[1585,] 1560  789
[1586,] 1559 1648
[1587,] 1648 1649
[1588,] 1649 1650
[1589,] 1650 1651
[1590,] 1651  790
[1591,] 1651  791
[1592,] 1650  792
[1593,] 1649  793
[1594,] 1648  794
[1595,] 1425 1652
[1596,] 1652 1653
[1597,] 1653 1654
[1598,] 1654 1655
[1599,] 1655 1656
[1600,] 1656 1657
[1601,] 1657 1658
[1602,] 1658 1659
[1603,] 1659 1660
[1604,] 1660 1661
[1605,] 1661  795
[1606,] 1661  796
[1607,] 1660  797
[1608,] 1659  798
[1609,] 1658 1662
[1610,] 1662  799
[1611,] 1662 1663
[1612,] 1663  800
[1613,] 1663  801
[1614,] 1657  802
[1615,] 1656  803
[1616,] 1655 1664
[1617,] 1664  804
[1618,] 1664 1665
[1619,] 1665 1666
[1620,] 1666 1667
[1621,] 1667  805
[1622,] 1667  806
[1623,] 1666 1668
[1624,] 1668  807
[1625,] 1668 1669
[1626,] 1669  808
[1627,] 1669  809
[1628,] 1665  810
[1629,] 1654  811
[1630,] 1653  812
[1631,] 1652 1670
[1632,] 1670  813
[1633,] 1670  814
[1634,] 1424 1671
[1635,] 1671 1672
[1636,] 1672  815
[1637,] 1672 1673
[1638,] 1673  816
[1639,] 1673  817
[1640,] 1671 1674
[1641,] 1674 1675
[1642,] 1675  818
[1643,] 1675 1676
[1644,] 1676  819
[1645,] 1676 1677
[1646,] 1677 1678
[1647,] 1678 1679
[1648,] 1679 1680
[1649,] 1680 1681
[1650,] 1681 1682
[1651,] 1682 1683
[1652,] 1683  820
[1653,] 1683  821
[1654,] 1682  822
[1655,] 1681  823
[1656,] 1680  824
[1657,] 1679  825
[1658,] 1678  826
[1659,] 1677  827
[1660,] 1674  828
[1661,] 1423 1684
[1662,] 1684 1685
[1663,] 1685 1686
[1664,] 1686  829
[1665,] 1686  830
[1666,] 1685  831
[1667,] 1684  832
[1668,] 1422 1687
[1669,] 1687 1688
[1670,] 1688 1689
[1671,] 1689 1690
[1672,] 1690 1691
[1673,] 1691 1692
[1674,] 1692 1693
[1675,] 1693  833
[1676,] 1693  834
[1677,] 1692  835
[1678,] 1691  836
[1679,] 1690  837
[1680,] 1689 1694
[1681,] 1694  838
[1682,] 1694 1695
[1683,] 1695  839
[1684,] 1695  840
[1685,] 1688  841
[1686,] 1687  842
[1687,]  879  843
[1688,]  854 1696
[1689,] 1696 1697
[1690,] 1697 1698
[1691,] 1698 1699
[1692,] 1699  844
[1693,] 1699  845
[1694,] 1698  846
[1695,] 1697  847
[1696,] 1696  848
[1697,]  853  849
[1698,]  851  850</code></pre>
<pre class="r"><code>n_tips &lt;- length(phy_tree$tip.label)

# let&#39;s color the branches
df_tree_color &lt;- tibble(phy_id = phy_tree$tip.label, color=rep(1,n_tips),
       umap = ifelse(phy_tree$tip.label %in% strains_umap,T,F),
       phy = ifelse(phy_tree$tip.label %in% strains_phy,T,F))


sum(df_tree_color$umap) # confirmed 100</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>sum(df_tree_color$phy) # confirmed 100</code></pre>
<pre><code>[1] 100</code></pre>
<pre class="r"><code># color
df_tree_color$color &lt;- ifelse(df_tree_color$umap &amp; df_tree_color$phy,4,# if in both umap and phy (blue)
                              ifelse(df_tree_color$phy, 3, # if in only phy (green)
                                     ifelse(df_tree_color$umap, 2, 1))) # if in only umap (red)
table(df_tree_color$color) # 1: black, 2: umap, 3: phy, 4: both</code></pre>
<pre><code>
  1   2   3   4 
733  17  97   3 </code></pre>
<pre class="r"><code># (1) Plot tree with umap-method strains indicated
plot.phylo(phy_tree, show.tip.label = F, cex=0.1,
           # align.tip.label = 2,
           # tip.color=df_tree_color$umap, 
           edge.width=1,
           # adj = c(1, 0.5),
           no.margin = T)
tiplabels(tip = which(df_tree_color$umap &gt; 0), pch = 19, cex = 2, col=&#39;maroon&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-10-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (2) Plot tree with umap-method strains indicated
plot.phylo(phy_tree, show.tip.label = F, cex=0.1,
           # align.tip.label = 2,
           # tip.color=df_tree_color$umap, 
           edge.width=1,
           # adj = c(1, 0.5),
           no.margin = T)
tiplabels(tip = which(df_tree_color$phy &gt; 0), pch = 19, cex = 2, col=&#39;skyblue&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-10-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot together with colors
plot.phylo(phy_tree, show.tip.label = F, cex=0.1,
           # align.tip.label = 2,
           # tip.color=df_tree_color$umap, 
           edge.width=1,
           # adj = c(1, 0.5),
           no.margin = T)
tiplabels(tip = which(df_tree_color$color == 2), pch = 19, cex = 2, col=&#39;maroon&#39;)
tiplabels(tip = which(df_tree_color$color == 3), pch = 19, cex = 2, col=&#39;skyblue&#39;)
tiplabels(tip = which(df_tree_color$color == 4), pch = 19, cex = 2, col=&#39;navy&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-10-3.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Calculate NRI and NTI</p>
<pre class="r"><code># making binary community matrix
umap = phy_tree$tip.label %in% strains_umap
phylogenetic = phy_tree$tip.label %in% strains_phy

# get community matrix with different species
select_different_species &lt;- function(n=100){
  phy_species &lt;- phylo_Biobank %&gt;% select(phy_id, species)
  unique_species_vec &lt;- unique(phylo_Biobank$species)  # unique species vector
  
  # randomly select n=100 different species
  diff_species_rand1 &lt;- sample(unique_species_vec, n)
  phy_species %&lt;&gt;% filter(species %in% diff_species_rand1) # get strains with those sampled species
  dim(phy_species)
  
  # randomly select 1 strain of each 100 species
  diff_species &lt;- phy_species[sample(1:dim(phy_species)[1]),] %&gt;% group_by(species) %&gt;% filter(row_number()==1)
  length(unique(diff_species$species)) # 100

  diff_species1 = phy_tree$tip.label %in% diff_species$phy_id
  return(diff_species1)
}

select_different_species(n=100)</code></pre>
<pre><code>  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE
 [13]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE
 [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE
 [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE
 [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [97] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE FALSE
[109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE
[121] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[133]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE  TRUE FALSE
[145] FALSE FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE
[157] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[169] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE
[181] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[193] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[205] FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE FALSE  TRUE FALSE FALSE FALSE
[217]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[229] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[241] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[253] FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[265]  TRUE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[277] FALSE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[289] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE
[301] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[313] FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE
[325] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE
[337]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[349] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE
[361] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[373] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE
[385] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE  TRUE
[397]  TRUE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE
[409]  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE
[421] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE
[433] FALSE FALSE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE  TRUE
[445] FALSE  TRUE FALSE  TRUE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE
[457] FALSE FALSE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[469] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[481] FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[493] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[505] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[517] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[529] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE  TRUE FALSE
[541] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE
[553] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE
[565]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[577] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[589] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[601]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[613] FALSE FALSE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE
[625] FALSE FALSE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE
[637] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[649] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[661] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[673] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[685] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[697] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE
[709] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[721] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[733] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[745] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[757] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[769] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[781] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[793] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE
[805] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE
[817]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[829]  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE
[841]  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE</code></pre>
<pre class="r"><code>set.seed(1)
dif_species1 = select_different_species(n=100)
dif_species2 = select_different_species(n=100)
dif_species3 = select_different_species(n=100)

# random 
set.seed(61)
random1 = phy_tree$tip.label %in% sample(phy_tree$tip.label, 100)
random2 = phy_tree$tip.label %in% sample(phy_tree$tip.label, 100)
random3 = phy_tree$tip.label %in% sample(phy_tree$tip.label, 100)

# bind it into community matrix
mat_com &lt;- rbind(umap, phylogenetic, dif_species1, dif_species2, dif_species3, random1, random2, random3)
colnames(mat_com) &lt;- phy_tree$tip.label

dim(mat_com)</code></pre>
<pre><code>[1]   8 850</code></pre>
<pre class="r"><code>phydist &lt;- cophenetic(phy_tree)
dim(phydist)</code></pre>
<pre><code>[1] 850 850</code></pre>
<pre class="r"><code>ses.mpd(mat_com, dis = phydist, null.model=&quot;taxa.labels&quot;)</code></pre>
<pre><code>             ntaxa   mpd.obs mpd.rand.mean mpd.rand.sd mpd.obs.rank  mpd.obs.z
umap            20 1.0420003      1.017092  0.07671684          591  0.3246805
phylogenetic   100 0.9642132      1.019526  0.02916655           41 -1.8964335
dif_species1   100 0.9873727      1.019430  0.02910822          144 -1.1013251
dif_species2   100 1.0136890      1.018833  0.02910586          411 -0.1767216
dif_species3   100 1.0161934      1.019563  0.02787694          431 -0.1208775
random1        100 1.0431256      1.019790  0.02800579          799  0.8332449
random2        100 1.0463130      1.018103  0.02910314          828  0.9693177
random3        100 1.0534477      1.019574  0.02712531          915  1.2487907
             mpd.obs.p runs
umap             0.591  999
phylogenetic     0.041  999
dif_species1     0.144  999
dif_species2     0.411  999
dif_species3     0.431  999
random1          0.799  999
random2          0.828  999
random3          0.915  999</code></pre>
<pre class="r"><code>NRI_table &lt;- ses.mpd(mat_com, dis = phydist, null.model=&quot;taxa.labels&quot;)
NRI_table &lt;- tibble::rownames_to_column(NRI_table, var=&quot;method&quot;)
write.xlsx(NRI_table, &quot;NRI_table.xlsx&quot;)

# The rows are the communities.  The first four columns should be pretty straight forward given the definition of the NRI from lecture.  To review:
# ntaxa Number of taxa in community
# mpd.obs Observed mean pairwise distance (MPD) in community
# mpd.rand.mean Mean MPD in null communities
# mpd.rand.sd Standard deviation of MPD in null communities
# mpd.obs.rank Rank of observed MPD vs. null communities
# mpd.obs.z Standardized effect size of MPD vs. null communities (equivalent to -NRI)

NTI_table &lt;- ses.mntd(mat_com, dis = phydist, null.model=&quot;taxa.labels&quot;)
NTI_table &lt;- tibble::rownames_to_column(NTI_table, var=&quot;method&quot;)
write.xlsx(NTI_table, &quot;NTI_table.xlsx&quot;)</code></pre>
<p>Histogram of species</p>
<pre class="r"><code>phy_species &lt;- phylo_Biobank %&gt;% select(phy_id, species)
hist_spe &lt;- phy_species %&gt;% group_by(species) %&gt;% summarize(frequency=n_distinct(phy_id))
histogram(hist_spe$frequency, xlab=&quot;Number of Strains in the same species&quot;, breaks=30)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-12-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="selection-of-strains" class="section level2">
<h2>5. Selection of strains</h2>
<ul>
<li>Selecting strains belonging to different species is the optimal strategy.</li>
<li>Randomly pick and choose the community with lowest NTI.</li>
</ul>
<pre class="r"><code># matrix
n &lt;- 1000
n_tips &lt;- length(phy_tree$tip.label)

mat_1000 &lt;- matrix(rep(-1,n*n_tips), nrow=n,ncol=n_tips)
dim(mat_1000)</code></pre>
<pre><code>[1] 1000  850</code></pre>
<pre class="r"><code>for (i in 1:n){
  mat_1000[i,] = select_different_species(n=95)
}

colnames(mat_1000) &lt;- phy_tree$tip.label
rownames(mat_1000) &lt;- paste0(&quot;set&quot;,1:n)
# head(mat_1000)

NTI_species_table &lt;- ses.mntd(mat_1000, dis = phydist, null.model = &quot;taxa.labels&quot;)
NTI_species_table &lt;- tibble::rownames_to_column(NTI_species_table, var=&quot;method&quot;)
NTI_species_table %&gt;% filter(mntd.obs.z == max(NTI_species_table$mntd.obs.z))</code></pre>
<pre><code>  method ntaxa  mntd.obs mntd.rand.mean mntd.rand.sd mntd.obs.rank mntd.obs.z
1 set323    95 0.1287507     0.04715426  0.009826457          1000    8.30375
  mntd.obs.p runs
1          1  999</code></pre>
<pre class="r"><code>set_number &lt;- NTI_species_table %&gt;% filter(mntd.obs.z == max(NTI_species_table$mntd.obs.z))
set_number &lt;- set_number$method
set_number</code></pre>
<pre><code>[1] &quot;set323&quot;</code></pre>
<pre class="r"><code>selected_strain_list &lt;- mat_1000[set_number,]
selected_strain_vec &lt;- names(selected_strain_list[selected_strain_list &gt; 0])

# get excel
colnames(phylo_Biobank)</code></pre>
<pre><code> [1] &quot;phy_id&quot;       &quot;seq_id&quot;       &quot;umapA1&quot;       &quot;umapA2&quot;       &quot;tiplabel&quot;    
 [6] &quot;msk_id&quot;       &quot;length_bp&quot;    &quot;phylum&quot;       &quot;class&quot;        &quot;order&quot;       
[11] &quot;family&quot;       &quot;genus&quot;        &quot;species&quot;      &quot;nuc_sequence&quot;</code></pre>
<pre class="r"><code>df_95strains &lt;- phylo_Biobank %&gt;% filter(phy_id %in% selected_strain_vec)
write.xlsx(df_95strains,&quot;Strain_selected_95_max_NTI.xlsx&quot;)

# plot UMAP
plot_UMAP(phylo_Biobank %&gt;% filter(phy_id %in% selected_strain_vec))</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-13-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot tree
df_tree_color &lt;- tibble(phy_id = phy_tree$tip.label, species = ifelse(phy_tree$tip.label %in% selected_strain_vec,T,F))

plot.phylo(phy_tree, show.tip.label = F, cex=0.1,
           # align.tip.label = 2,
           # tip.color=df_tree_color$umap, 
           edge.width=1,
           # adj = c(1, 0.5),
           no.margin = T)
tiplabels(tip = which(df_tree_color$species &gt; 0), pch = 19, cex = 2, col=&#39;green&#39;)</code></pre>
<p><img src="figure/Phylogenetic_strain_selection.Rmd/unnamed-chunk-13-2.png" width="1056" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] picante_1.8.2      nlme_3.1-149       FCPS_1.3.0         phangorn_2.7.1    
 [5] cluster_2.1.0      ape_5.5            openxlsx_4.2.3     devtools_2.4.0    
 [9] usethis_2.0.1      gtools_3.8.2       reshape2_1.4.4     readxl_1.3.1      
[13] magrittr_2.0.1     forcats_0.5.1      stringr_1.4.0      purrr_0.3.4       
[17] readr_1.4.0        tidyr_1.1.3        tibble_3.0.4       tidyverse_1.3.1   
[21] vegan_2.5-7        lattice_0.20-41    permute_0.9-5      RColorBrewer_1.1-2
[25] ggplot2_3.3.5      dplyr_1.0.5        workflowr_1.6.2   

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0         ellipsis_0.3.2           mclust_5.4.7            
 [4] rprojroot_2.0.2          fs_1.5.0                 rstudioapi_0.13         
 [7] farver_2.1.0             remotes_2.4.0            fansi_0.4.2             
[10] lubridate_1.7.10         xml2_1.3.2               codetools_0.2-16        
[13] splines_4.0.3            cachem_1.0.4             knitr_1.37              
[16] pkgload_1.2.1            jsonlite_1.7.2           broom_0.7.9             
[19] dbplyr_2.1.1             compiler_4.0.3           httr_1.4.2              
[22] backports_1.2.1          assertthat_0.2.1         Matrix_1.2-18           
[25] fastmap_1.1.0            cli_3.0.1                later_1.2.0             
[28] htmltools_0.5.1.1        prettyunits_1.1.1        tools_4.0.3             
[31] igraph_1.2.6             gtable_0.3.0             glue_1.4.2              
[34] DataVisualizations_1.2.2 fastmatch_1.1-3          Rcpp_1.0.8.3            
[37] cellranger_1.1.0         jquerylib_0.1.4          vctrs_0.3.8             
[40] crosstalk_1.1.1          xfun_0.29                ps_1.6.0                
[43] testthat_3.0.2           rvest_1.0.1              lifecycle_1.0.0         
[46] MASS_7.3-53              scales_1.1.1             hms_1.1.0               
[49] promises_1.2.0.1         parallel_4.0.3           yaml_2.2.1              
[52] memoise_2.0.0            sass_0.4.0               stringi_1.5.3           
[55] highr_0.9                desc_1.3.0               pkgbuild_1.2.0          
[58] zip_2.1.1                rlang_0.4.10             pkgconfig_2.0.3         
[61] rgl_0.107.10             evaluate_0.14            htmlwidgets_1.5.3       
[64] labeling_0.4.2           tidyselect_1.1.1         processx_3.5.1          
[67] plyr_1.8.6               R6_2.5.0                 generics_0.1.0          
[70] DBI_1.1.1                pillar_1.6.0             haven_2.4.1             
[73] whisker_0.4              withr_2.4.2              mgcv_1.8-33             
[76] sp_1.4-5                 modelr_0.1.8             crayon_1.4.1            
[79] utf8_1.1.4               rmarkdown_2.9            grid_4.0.3              
[82] callr_3.7.0              git2r_0.28.0             reprex_2.0.0            
[85] digest_0.6.27            httpuv_1.6.0             munsell_0.5.0           
[88] bslib_0.2.5.1            sessioninfo_1.1.1        quadprog_1.5-8          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
